<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="composition-plots.html">
<link rel="next" href="core-microbiota.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>2.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>2.9</b> Variablity</a></li>
<li class="chapter" data-level="2.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>2.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>2.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>3</b> Alpha diversities</a><ul>
<li class="chapter" data-level="3.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>3.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="3.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>3.2</b> Diversities</a><ul>
<li class="chapter" data-level="3.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>3.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="3.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>3.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>4</b> Composition plots</a><ul>
<li class="chapter" data-level="4.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>4.1</b> Barplot counts</a></li>
<li class="chapter" data-level="4.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="4.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>4.2.1</b> Barplot customize</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>5</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="5.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>5.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="5.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>5.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>5.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="5.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>5.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="5.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>6</b> Core microbiota</a><ul>
<li class="chapter" data-level="6.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>6.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="6.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>6.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="6.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>6.3</b> Core visualization</a><ul>
<li class="chapter" data-level="6.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>6.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>7</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="7.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>7.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="7.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>7.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="7.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>7.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="7.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>7.2.2</b> igraph network</a></li>
<li class="chapter" data-level="7.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>7.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>7.3</b> Network properties</a><ul>
<li class="chapter" data-level="7.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>7.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="7.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>7.3.2</b> Network plot</a></li>
<li class="chapter" data-level="7.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>7.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>8</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="8.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>8.1</b> Load example data</a></li>
<li class="chapter" data-level="8.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="8.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="8.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>8.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="8.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>8.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>9</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="9.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>9.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="9.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>9.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="9.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>9.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="10.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>10.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="10.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>10.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="10.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>10.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="10.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>10.1.3</b> Rarity</a></li>
<li class="chapter" data-level="10.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>10.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>10.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="10.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>10.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="10.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>10.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="10.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>10.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>11</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="11.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>11.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="11.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>11.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>12</b> Further exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>12.1</b> Compositionality</a></li>
<li class="chapter" data-level="12.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>12.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="beta-diversity-metrics" class="section level1">
<h1><span class="header-section-number">5</span> Beta diversity metrics</h1>
<p>Beta-diversity: Measures for differences between samples from different groups to identify if there are differences in the overall community composition and structure.</p>
<p><strong>Load packages and data</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome) <span class="co"># data analysis and visualisation</span>
<span class="kw">library</span>(phyloseq) <span class="co"># also the basis of data object. Data analysis and visualisation</span>
<span class="kw">library</span>(RColorBrewer) <span class="co"># nice color options</span>
<span class="kw">library</span>(ggpubr) <span class="co"># publication quality figures, based on ggplot2</span>
<span class="kw">library</span>(dplyr) <span class="co"># data handling  </span></code></pre></div>
<p>For more information:</p>
<p><a href="http://ordination.okstate.edu/overview.htm">Ordination</a>.</p>
<p><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531">Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible</a>.</p>
<p><a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0237-y">Normalisation and data transformation</a>.</p>
<p><a href="http://www.davidzeleny.net/anadat-r/doku.php/en:ordination">What is Constrained and Unconstrained Ordination</a>.</p>
<p><a href="https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full">Microbiome Datasets Are Compositional: And This Is Not Optional</a></p>
<p><a href="http://www.nrcresearchpress.com/doi/full/10.1139/cjm-2015-0821#.WvIJZIiFM2w">Compositional analysis: a valid approach to analyze microbiome high-throughput sequencing data</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># read non rarefied data</span>
ps1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps.ng.tax.rds&quot;</span>)

<span class="co"># read rarefied data</span>
ps0.rar.rds &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps0.rar.rds&quot;</span>)
<span class="co"># use print option to see the data saved as phyloseq object.</span></code></pre></div>
<div id="phylogenetic-beta-diversity-metrics" class="section level2">
<h2><span class="header-section-number">5.1</span> Phylogenetic beta-diversity metrics</h2>
<div id="unweighted-unifrac" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Unweighted Unifrac</h3>
<p>Unweighted Unifrac is based on presence/absence of different taxa and abundance is not important. However, it is sensitive to the sequencing depth. If a sample is sequenced more than the others then it may have many OTUs (most of them unique) consequently affecting the unifrac dissimilarity estimation.</p>
<p>Usually, using subOTU/ASV approaches many singletons/OTUs with very low reads are discarded. If you have you won data you try the following code. For data from NG-tax we will skip this step as we have no singletons.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># if we remove OTUs that are detected atleast 10 times in 5% of the samples</span>
ps0.rar.filtered &lt;-<span class="st"> </span><span class="kw">core</span>(ps0.rar.rds, <span class="dt">detection =</span> <span class="dv">10</span>, <span class="dt">prevalence =</span> <span class="fl">0.05</span>)

<span class="kw">summarize_phyloseq</span>(ps0.rar.filtered)

<span class="co"># we reduce the sparsity considerably. </span></code></pre></div>
<p><em>For data from OTU picking</em><br />
Since the data used here consists of different body sites with distinct biological properties, the results of ordination do not change a lot by filtering “rare” OTUs. Once again, knowing the biology of your samples and making choices rationally and documenting them is crucial.<br />
Feel free to use the OTU-picking strategy phyloseq object to investigate yourself.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ordu.unwt.uni &lt;-<span class="st"> </span><span class="kw">ordinate</span>(ps0.rar.rds, <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="dt">weighted=</span>F)

<span class="co"># check for Eigen values </span>
<span class="co"># barplot(ordu.unwt.uni$values$Eigenvalues[1:10])</span>

unwt.unifrac &lt;-<span class="st"> </span><span class="kw">plot_ordination</span>(ps0.rar.rds, 
                                     ordu.unwt.uni, <span class="dt">color=</span><span class="st">&quot;scientific_name&quot;</span>) 
unwt.unifrac &lt;-<span class="st"> </span>unwt.unifrac <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Unweighted UniFrac&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">2</span>)
unwt.unifrac &lt;-<span class="st"> </span>unwt.unifrac <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="st">&quot;Location&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Set2&quot;</span>)
<span class="kw">print</span>(unwt.unifrac)</code></pre></div>
<p><img src="05-MAW-PIV_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Try repeating the above ordination using non-filtered phyloseq object.</p>
</div>
<div id="weighted-unifrac" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Weighted Unifrac</h3>
<p>Weighted Unifrac will consider the abundances of different taxa.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.rel &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">transform</span>(ps1, <span class="st">&quot;compositional&quot;</span>)

ordu.wt.uni &lt;-<span class="st"> </span><span class="kw">ordinate</span>(ps1.rel , <span class="st">&quot;PCoA&quot;</span>, <span class="st">&quot;unifrac&quot;</span>, <span class="dt">weighted=</span>T)

<span class="co"># check for Eigen values </span>
<span class="co"># barplot(ordu.unwt.uni$values$Eigenvalues[1:10])</span>

wt.unifrac &lt;-<span class="st"> </span><span class="kw">plot_ordination</span>(ps1.rel, 
                                     ordu.wt.uni, <span class="dt">color=</span><span class="st">&quot;scientific_name&quot;</span>) 
wt.unifrac &lt;-<span class="st"> </span>wt.unifrac <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Weighted UniFrac&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">2</span>)
wt.unifrac &lt;-<span class="st"> </span>wt.unifrac <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="st">&quot;Location&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Set2&quot;</span>)
<span class="kw">print</span>(wt.unifrac)</code></pre></div>
<p><img src="05-MAW-PIV_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(wt.unifrac <span class="op">+</span><span class="st"> </span><span class="kw">stat_ellipse</span>())</code></pre></div>
<p><img src="05-MAW-PIV_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
<p>The figure brings forward an important characteristics of microbiome data called the ‘Horse-shoe effect’. An investigation and explaination for this can be found in the article by Morton JT., et al. 2017 <a href="http://msystems.asm.org/content/2/1/e00166-16">Uncovering the Horseshoe Effect in Microbial Analyses</a>.</p>
<p>You can repeating this analysis with phyloseq object from OTU-picking approach.</p>
<p>Another important aspect regarding weighted unifrac is its property of having heavier weights for abundant taxa. To detect changes in moderately abundant lineages, an extenstion called generalized (UniFrac distance)(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413390/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413390/</a>) has been developed. In this test data, we expect sufficient biological variation in composition between sites and hence, we do not apply GUniFrac.</p>
<p><strong>To reiterate:</strong> It is crucial to understand the biological features of the samples. Although these are exploratory approaches, it is important to differentiate between biological signal and technical artifacts.</p>
</div>
</div>
<div id="population-level-density-landscapes" class="section level2">
<h2><span class="header-section-number">5.2</span> Population-level Density landscapes</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">plot_landscape</span>(ps1.rel, 
                    <span class="st">&quot;NMDS&quot;</span>, 
                    <span class="st">&quot;bray&quot;</span>, 
                    <span class="dt">col =</span> <span class="st">&quot;scientific_name&quot;</span>) <span class="op">+</span>
<span class="st">       </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="kw">paste</span>(<span class="st">&quot;NMDS / Bray-Curtis&quot;</span>))   

p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Dark2&quot;</span>)<span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;#e0ecf4&quot;</span>, <span class="dt">high =</span> <span class="st">&quot;#6e016b&quot;</span>) </code></pre></div>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p </code></pre></div>
<p><img src="05-MAW-PIV_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>Bray-Curtis dissimilarity does not consider phylogenetic relationships between OTUs. There are several distance methods and a list can be obtained by typying <code>?distanceMethodList</code> in the console pane.</p>
<p>Section on multivariate analysis will be discussed on Day3.</p>
</div>
<div id="permanova" class="section level2">
<h2><span class="header-section-number">5.3</span> PERMANOVA</h2>
<p>Permutational multivariate analysis of variance <a href="https://onlinelibrary.wiley.com/doi/10.1002/9781118445112.stat07841">further reading</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vegan)</code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.5-2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">metadf &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">sample_data</span>(ps1.rel))

unifrac.dist &lt;-<span class="st"> </span><span class="kw">UniFrac</span>(ps1.rel, 
                        <span class="dt">weighted =</span> <span class="ot">TRUE</span>, 
                        <span class="dt">normalized =</span> <span class="ot">TRUE</span>,  
                        <span class="dt">parallel =</span> <span class="ot">FALSE</span>, 
                        <span class="dt">fast =</span> <span class="ot">TRUE</span>)

permanova &lt;-<span class="st"> </span><span class="kw">adonis</span>(unifrac.dist <span class="op">~</span><span class="st"> </span>scientific_name, <span class="dt">data =</span> metadf)

permanova</code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = unifrac.dist ~ scientific_name, data = metadf) 
## 
## Permutation: free
## Number of permutations: 999
## 
## Terms added sequentially (first to last)
## 
##                  Df SumsOfSqs MeanSqs F.Model     R2 Pr(&gt;F)    
## scientific_name   3    71.912 23.9705  376.03 0.7059  0.001 ***
## Residuals       470    29.960  0.0637         0.2941           
## Total           473   101.872                 1.0000           
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="checking-the-homogeneity-condition" class="section level2">
<h2><span class="header-section-number">5.4</span> Checking the homogeneity condition</h2>
<p>Type <code>?betadisper</code> in R console for more information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps.disper &lt;-<span class="st"> </span><span class="kw">betadisper</span>(unifrac.dist, metadf<span class="op">$</span>scientific_name)
<span class="kw">permutest</span>(ps.disper, <span class="dt">pairwise =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##            Df Sum Sq Mean Sq      F N.Perm Pr(&gt;F)    
## Groups      3 1.0014 0.33381 31.889    999  0.001 ***
## Residuals 470 4.9200 0.01047                         
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##                          human gut metagenome human oral metagenome
## human gut metagenome                                     1.0000e-03
## human oral metagenome              8.3027e-05                      
## human skin metagenome              2.8177e-06            2.4596e-02
## human vaginal metagenome           3.1013e-18            1.4078e-10
##                          human skin metagenome human vaginal metagenome
## human gut metagenome                1.0000e-03                    0.001
## human oral metagenome               2.9000e-02                    0.001
## human skin metagenome                                             0.014
## human vaginal metagenome            7.0275e-03</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 16299)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] methods   stats     graphics  grDevices utils     datasets  base     
## 
## other attached packages:
##  [1] vegan_2.5-2          lattice_0.20-35      permute_0.9-4       
##  [4] dplyr_0.7.5          ggpubr_0.1.6         magrittr_1.5        
##  [7] RColorBrewer_1.1-2   microbiome_1.1.10013 ggplot2_2.2.1.9000  
## [10] phyloseq_1.22.3     
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_0.2.4    xfun_0.1            reshape2_1.4.3     
##  [4] purrr_0.2.4         splines_3.4.4       rhdf5_2.22.0       
##  [7] colorspace_1.3-2    htmltools_0.3.6     stats4_3.4.4       
## [10] mgcv_1.8-23         yaml_2.1.19         survival_2.41-3    
## [13] rlang_0.2.0         pillar_1.2.2        withr_2.1.2        
## [16] glue_1.2.0          BiocGenerics_0.24.0 bindrcpp_0.2.2     
## [19] foreach_1.4.4       plyr_1.8.4          bindr_0.1.1        
## [22] stringr_1.3.1       zlibbioc_1.24.0     Biostrings_2.46.0  
## [25] munsell_0.4.3       gtable_0.2.0        codetools_0.2-15   
## [28] evaluate_0.10.1     labeling_0.3        Biobase_2.38.0     
## [31] knitr_1.20          IRanges_2.12.0      biomformat_1.6.0   
## [34] parallel_3.4.4      Rcpp_0.12.17        backports_1.1.2    
## [37] scales_0.5.0        S4Vectors_0.16.0    jsonlite_1.5       
## [40] XVector_0.18.0      digest_0.6.15       Rtsne_0.13         
## [43] stringi_1.2.2       bookdown_0.7        grid_3.4.4         
## [46] ade4_1.7-11         rprojroot_1.3-2     tools_3.4.4        
## [49] lazyeval_0.2.1      tibble_1.4.2        cluster_2.0.6      
## [52] tidyr_0.8.1         ape_5.1             pkgconfig_2.0.1    
## [55] MASS_7.3-49         Matrix_1.2-12       data.table_1.11.2  
## [58] assertthat_0.2.0    rmarkdown_1.9       iterators_1.0.9    
## [61] R6_2.2.2            multtest_2.34.0     igraph_1.2.1       
## [64] nlme_3.1-131.1      compiler_3.4.4</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="composition-plots.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="core-microbiota.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
