<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="alpha-diversities.html">
<link rel="next" href="beta-diversity-metrics.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>2.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>2.9</b> Variablity</a></li>
<li class="chapter" data-level="2.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>2.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>2.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>3</b> Alpha diversities</a><ul>
<li class="chapter" data-level="3.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>3.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="3.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>3.2</b> Diversities</a><ul>
<li class="chapter" data-level="3.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>3.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="3.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>3.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>4</b> Composition plots</a><ul>
<li class="chapter" data-level="4.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>4.1</b> Barplot counts</a></li>
<li class="chapter" data-level="4.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="4.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>4.2.1</b> Barplot customize</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>5</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="5.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>5.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="5.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>5.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>5.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="5.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>5.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="5.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>6</b> Core microbiota</a><ul>
<li class="chapter" data-level="6.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>6.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="6.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>6.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="6.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>6.3</b> Core visualization</a><ul>
<li class="chapter" data-level="6.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>6.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>7</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="7.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>7.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="7.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>7.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="7.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>7.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="7.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>7.2.2</b> igraph network</a></li>
<li class="chapter" data-level="7.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>7.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>7.3</b> Network properties</a><ul>
<li class="chapter" data-level="7.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>7.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="7.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>7.3.2</b> Network plot</a></li>
<li class="chapter" data-level="7.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>7.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>8</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="8.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>8.1</b> Load example data</a></li>
<li class="chapter" data-level="8.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="8.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="8.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>8.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="8.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>8.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>9</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="9.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>9.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="9.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>9.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="9.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>9.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="10.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>10.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="10.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>10.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="10.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>10.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="10.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>10.1.3</b> Rarity</a></li>
<li class="chapter" data-level="10.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>10.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>10.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="10.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>10.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="10.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>10.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="10.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>10.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>11</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="11.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>11.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="11.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>11.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>12</b> Further exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>12.1</b> Compositionality</a></li>
<li class="chapter" data-level="12.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>12.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="composition-plots" class="section level1">
<h1><span class="header-section-number">4</span> Composition plots</h1>
<p>Barplots are a one way of visualising the composition of your samples.</p>
<p>We will use the filtered phyloseq object from <strong>Set-up and Pre-processing</strong> section.</p>
<p><strong>Load packages</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome) <span class="co"># data analysis and visualisation</span>
<span class="kw">library</span>(phyloseq) <span class="co"># also the basis of data object. Data analysis and visualisation</span>
<span class="kw">library</span>(RColorBrewer) <span class="co"># nice color options</span>
<span class="kw">library</span>(ggpubr) <span class="co"># publication quality figures, based on ggplot2</span>
<span class="kw">library</span>(dplyr) <span class="co"># data handling  </span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps.ng.tax.rds&quot;</span>)

<span class="co"># use print option to see the data saved as phyloseq object.</span>

<span class="kw">print</span>(ps1)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 4710 taxa and 474 samples ]
## sample_data() Sample Data:       [ 474 samples by 30 sample variables ]
## tax_table()   Taxonomy Table:    [ 4710 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 4710 tips and 4709 internal nodes ]</code></pre>
<div id="barplot-counts" class="section level2">
<h2><span class="header-section-number">4.1</span> Barplot counts</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.com &lt;-<span class="st"> </span>ps1

<span class="co"># if you have dada2/deblur output and sequences as taxa names, then you can change them as follows</span>
<span class="kw">taxa_names</span>(ps1.com) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ASV_&quot;</span>, <span class="kw">rownames</span>(<span class="kw">tax_table</span>(ps1.com)))

<span class="co"># We need to set Palette</span>
taxic &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(ps1.com<span class="op">@</span>tax_table)  <span class="co"># this will help in setting large color options</span>

<span class="co">#colourCount = length(unique(taxic$Family))  #define number of variable colors based on number of Family (change the level accordingly to phylum/class/order)</span>
<span class="co">#getPalette = colorRampPalette(brewer.pal(12, &quot;Paired&quot;))  # change the palette as well as the number of colors will change according to palette.</span>

taxic<span class="op">$</span>OTU &lt;-<span class="st"> </span><span class="kw">rownames</span>(taxic)  <span class="co"># Add the OTU ids from OTU table into the taxa table at the end.</span>
<span class="kw">colnames</span>(taxic)  <span class="co"># You can see that we now have extra taxonomy levels.</span></code></pre></div>
<pre><code>## [1] &quot;Domain&quot; &quot;Phylum&quot; &quot;Class&quot;  &quot;Order&quot;  &quot;Family&quot; &quot;Genus&quot;  &quot;OTU&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">taxmat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(taxic)  <span class="co"># convert it into a matrix.</span>
new.tax &lt;-<span class="st"> </span><span class="kw">tax_table</span>(taxmat)  <span class="co"># convert into phyloseq compatible file.</span>
<span class="kw">tax_table</span>(ps1.com) &lt;-<span class="st"> </span>new.tax  <span class="co"># incroporate into phyloseq Object</span>


<span class="co"># now edit the unclassified taxa</span>
<span class="kw">tax_table</span>(ps1.com)[<span class="kw">tax_table</span>(ps1.com)[, <span class="st">&quot;Family&quot;</span>] <span class="op">==</span><span class="st"> &quot;&quot;</span>, <span class="st">&quot;Family&quot;</span>] &lt;-<span class="st"> &quot;Unclassified family&quot;</span>

<span class="co"># it would be nice to have the Taxonomic names in italics.</span>
<span class="co"># for that we set this</span>

guide_italics &lt;-<span class="st"> </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">label.theme =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">15</span>, 
    <span class="dt">face =</span> <span class="st">&quot;italic&quot;</span>, <span class="dt">colour =</span> <span class="st">&quot;Black&quot;</span>, <span class="dt">angle =</span> <span class="dv">0</span>)))


## Now we need to plot at family level, we can do it as follows:

<span class="co"># first remove the phy_tree</span>

ps1.com<span class="op">@</span>phy_tree &lt;-<span class="st"> </span><span class="ot">NULL</span>

<span class="co"># Second merge at family level </span>

ps1.com.fam &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">aggregate_taxa</span>(ps1.com, <span class="st">&quot;Family&quot;</span>, <span class="dt">top =</span> <span class="dv">10</span>)

plot.composition.COuntAbun &lt;-<span class="st"> </span><span class="kw">plot_composition</span>(ps1.com.fam) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="st">&quot;Family&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Relative abundance&quot;</span>) <span class="op">+</span><span class="st"> </span>guide_italics <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">18</span>))
  
plot.composition.COuntAbun</code></pre></div>
<p><img src="04-MAW-PIII_files/figure-html/unnamed-chunk-3-1.png" width="1920" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#ggsave(&quot;./Test_Outputfiles/Family_barplot_CountAbundance.pdf&quot;, height = 6, width = 8)</span></code></pre></div>
<p>This plot is based on the reads per sample. In the next step, we plot the relative abundance.</p>
</div>
<div id="barplot-relative-abundance" class="section level2">
<h2><span class="header-section-number">4.2</span> Barplot relative abundance</h2>
<p>Make it relative abundance</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># the previous pseq object ps1.com.fam is only counts.</span>

<span class="co"># Use traqnsform function of microbiome to convert it to rel abun.</span>
ps1.com.fam.rel &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">transform</span>(ps1.com.fam, <span class="st">&quot;compositional&quot;</span>)

plot.composition.relAbun &lt;-<span class="st"> </span><span class="kw">plot_composition</span>(ps1.com.fam.rel, 
                                             <span class="dt">sample.sort =</span> <span class="st">&quot;scientific_name&quot;</span>, 
                                             <span class="dt">x.label =</span> <span class="st">&quot;env_material&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_brewer</span>(<span class="st">&quot;Family&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Relative abundance&quot;</span>) <span class="op">+</span><span class="st"> </span>guide_italics <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">18</span>))
  
plot.composition.relAbun</code></pre></div>
<p><img src="04-MAW-PIII_files/figure-html/unnamed-chunk-4-1.png" width="1920" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#ggsave(&quot;./figures/Family_barplot_RelAbundance.pdf&quot;, height = 6, width = 8)</span></code></pre></div>
<div id="barplot-customize" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Barplot customize</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.com &lt;-<span class="st"> </span>plot.composition.relAbun<span class="op">$</span>data
<span class="kw">colnames</span>(data.com)</code></pre></div>
<pre><code>## [1] &quot;OTU&quot;       &quot;Sample&quot;    &quot;Abundance&quot; &quot;xlabel&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p.com &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data.com, <span class="kw">aes</span>(<span class="dt">x =</span> Sample, <span class="dt">y =</span> Abundance, <span class="dt">fill =</span> OTU))
p.com &lt;-<span class="st"> </span>p.com <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;stack&quot;</span>, <span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>)
p.com &lt;-<span class="st"> </span>p.com <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_discrete</span>(<span class="dt">labels =</span> data.com<span class="op">$</span>xlabel, <span class="dt">breaks =</span> data.com<span class="op">$</span>Sample)
p.com &lt;-<span class="st"> </span>p.com <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(<span class="op">~</span>xlabel, <span class="dt">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()
p.com &lt;-<span class="st"> </span>p.com <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_brewer</span>(<span class="st">&quot;Family&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Paired&quot;</span>) 
p.com &lt;-<span class="st"> </span>p.com <span class="op">+</span><span class="st"> </span><span class="kw">rremove</span>(<span class="st">&quot;x.text&quot;</span>) 

<span class="kw">ggsave</span>(<span class="st">&quot;./figures/Composition plots.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="dv">6</span>)</code></pre></div>
<p>For more information <a href="http://microbiome.github.io/microbiome/Composition.html">Microbiome tutorial</a><br />
## Heatmaps<br />
These are a good alternative to barplots (if done right).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># base plot</span>
p.heat &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data.com, <span class="kw">aes</span>(<span class="dt">x =</span> Sample, <span class="dt">y =</span> OTU)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> Abundance)) 

<span class="co"># Change color</span>
p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_distiller</span>(<span class="st">&quot;Abundance&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;RdYlBu&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() 

<span class="co"># Make bacterial names italics</span>
p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&#39;black&#39;</span>, 
                                                    <span class="dt">size =</span> <span class="dv">10</span>, 
                                                    <span class="dt">face =</span> <span class="st">&#39;italic&#39;</span>)) 
<span class="co"># Make seperate samples based on main varaible</span>
p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(<span class="op">~</span>xlabel, 
                              <span class="dt">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">rremove</span>(<span class="st">&quot;x.text&quot;</span>) 

p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Family&quot;</span>)

<span class="co">#Clean the x-axis</span>
p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
                     <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>(),
                     <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>()) 

<span class="co"># Clean the facet label box</span>
p.heat &lt;-<span class="st"> </span>p.heat <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.key =</span> <span class="kw">element_blank</span>(), 
                     <span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;white&quot;</span>))

<span class="kw">print</span>(p.heat)</code></pre></div>
<p><img src="04-MAW-PIII_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;./figures/Heatmap.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="dv">6</span>)


<span class="co"># + geom_text(aes(label = round(Abundance)), size = 0.4)</span></code></pre></div>
<p><strong>Extra</strong></p>
<p>Following is an example of customizing the plot using ggpubr.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># we use count data at family level from the barplot for counts</span>
ps_df &lt;-<span class="st"> </span>microbiomeutilities<span class="op">::</span><span class="kw">phy_to_ldf</span>(ps1.com.fam, <span class="dt">transform.counts =</span> <span class="st">&quot;compositional&quot;</span>)</code></pre></div>
<pre><code>## An additonal column Sam_rep with sample names is created for reference purpose</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colnames</span>(ps_df)</code></pre></div>
<pre><code>##  [1] &quot;OTUID&quot;                       &quot;Family&quot;                     
##  [3] &quot;OTU&quot;                         &quot;Sam_rep&quot;                    
##  [5] &quot;Abundance&quot;                   &quot;BarcodeSequence&quot;            
##  [7] &quot;LinkerPrimerSequence&quot;        &quot;run_prefix&quot;                 
##  [9] &quot;body_habitat&quot;                &quot;body_product&quot;               
## [11] &quot;body_site&quot;                   &quot;bodysite&quot;                   
## [13] &quot;dna_extracted&quot;               &quot;elevation&quot;                  
## [15] &quot;env&quot;                         &quot;env_biome&quot;                  
## [17] &quot;env_feature&quot;                 &quot;env_material&quot;               
## [19] &quot;env_package&quot;                 &quot;geo_loc_name&quot;               
## [21] &quot;host_common_name&quot;            &quot;host_scientific_name&quot;       
## [23] &quot;host_subject_id&quot;             &quot;host_taxid&quot;                 
## [25] &quot;latitude&quot;                    &quot;longitude&quot;                  
## [27] &quot;physical_specimen_location&quot;  &quot;physical_specimen_remaining&quot;
## [29] &quot;psn&quot;                         &quot;public&quot;                     
## [31] &quot;sample_type&quot;                 &quot;scientific_name&quot;            
## [33] &quot;sequencecenter&quot;              &quot;title&quot;                      
## [35] &quot;Description&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># this data.frame can be used to customize several plots.  </span>

<span class="co"># example boxplot at family level</span>

p.box &lt;-<span class="st"> </span><span class="kw">ggstripchart</span>(ps_df, <span class="st">&quot;scientific_name&quot;</span>, <span class="st">&quot;Abundance&quot;</span>, 
                      <span class="dt">facet.by =</span> <span class="st">&quot;Family&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;body_product&quot;</span>,
                      <span class="dt">palette =</span> <span class="st">&quot;jco&quot;</span>
                      )

p.box <span class="op">+</span><span class="st"> </span><span class="kw">rremove</span>(<span class="st">&quot;x.text&quot;</span>)</code></pre></div>
<p><img src="04-MAW-PIII_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 16299)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] methods   stats     graphics  grDevices utils     datasets  base     
## 
## other attached packages:
## [1] bindrcpp_0.2.2       dplyr_0.7.5          ggpubr_0.1.6        
## [4] magrittr_1.5         RColorBrewer_1.1-2   microbiome_1.1.10013
## [7] ggplot2_2.2.1.9000   phyloseq_1.22.3     
## 
## loaded via a namespace (and not attached):
##  [1] ggrepel_0.8.0              Rcpp_0.12.17              
##  [3] ape_5.1                    lattice_0.20-35           
##  [5] tidyr_0.8.1                Biostrings_2.46.0         
##  [7] assertthat_0.2.0           rprojroot_1.3-2           
##  [9] digest_0.6.15              foreach_1.4.4             
## [11] R6_2.2.2                   plyr_1.8.4                
## [13] backports_1.1.2            stats4_3.4.4              
## [15] evaluate_0.10.1            pillar_1.2.2              
## [17] zlibbioc_1.24.0            rlang_0.2.0               
## [19] lazyeval_0.2.1             data.table_1.11.2         
## [21] vegan_2.5-2                S4Vectors_0.16.0          
## [23] Matrix_1.2-12              rmarkdown_1.9             
## [25] labeling_0.3               splines_3.4.4             
## [27] Rtsne_0.13                 stringr_1.3.1             
## [29] pheatmap_1.0.10            igraph_1.2.1              
## [31] munsell_0.4.3              compiler_3.4.4            
## [33] xfun_0.1                   pkgconfig_2.0.1           
## [35] BiocGenerics_0.24.0        multtest_2.34.0           
## [37] mgcv_1.8-23                htmltools_0.3.6           
## [39] biomformat_1.6.0           tidyselect_0.2.4          
## [41] gridExtra_2.3              tibble_1.4.2              
## [43] bookdown_0.7               IRanges_2.12.0            
## [45] codetools_0.2-15           viridisLite_0.3.0         
## [47] permute_0.9-4              withr_2.1.2               
## [49] MASS_7.3-49                grid_3.4.4                
## [51] nlme_3.1-131.1             jsonlite_1.5              
## [53] gtable_0.2.0               scales_0.5.0              
## [55] stringi_1.2.2              XVector_0.18.0            
## [57] reshape2_1.4.3             viridis_0.5.1             
## [59] ggsci_2.9                  iterators_1.0.9           
## [61] tools_3.4.4                microbiomeutilities_0.99.0
## [63] ade4_1.7-11                Biobase_2.38.0            
## [65] glue_1.2.0                 purrr_0.2.4               
## [67] parallel_3.4.4             survival_2.41-3           
## [69] yaml_2.1.19                colorspace_1.3-2          
## [71] rhdf5_2.22.0               cluster_2.0.6             
## [73] knitr_1.20                 bindr_0.1.1</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="alpha-diversities.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="beta-diversity-metrics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
