<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="beta-diversity-metrics.html">
<link rel="next" href="inference-of-microbial-ecological-networks.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>2.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>2.9</b> Variablity</a></li>
<li class="chapter" data-level="2.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>2.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>2.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>3</b> Alpha diversities</a><ul>
<li class="chapter" data-level="3.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>3.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="3.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>3.2</b> Diversities</a><ul>
<li class="chapter" data-level="3.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>3.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="3.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>3.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>4</b> Composition plots</a><ul>
<li class="chapter" data-level="4.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>4.1</b> Barplot counts</a></li>
<li class="chapter" data-level="4.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="4.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>4.2.1</b> Barplot customize</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>5</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="5.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>5.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="5.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>5.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>5.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="5.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>5.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="5.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>6</b> Core microbiota</a><ul>
<li class="chapter" data-level="6.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>6.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="6.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>6.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="6.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>6.3</b> Core visualization</a><ul>
<li class="chapter" data-level="6.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>6.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>7</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="7.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>7.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="7.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>7.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="7.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>7.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="7.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>7.2.2</b> igraph network</a></li>
<li class="chapter" data-level="7.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>7.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>7.3</b> Network properties</a><ul>
<li class="chapter" data-level="7.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>7.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="7.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>7.3.2</b> Network plot</a></li>
<li class="chapter" data-level="7.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>7.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>8</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="8.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>8.1</b> Load example data</a></li>
<li class="chapter" data-level="8.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="8.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="8.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>8.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="8.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>8.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>9</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="9.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>9.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="9.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>9.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="9.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>9.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="10.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>10.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="10.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>10.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="10.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>10.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="10.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>10.1.3</b> Rarity</a></li>
<li class="chapter" data-level="10.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>10.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>10.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="10.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>10.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="10.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>10.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="10.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>10.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>11</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="11.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>11.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="11.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>11.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>12</b> Further exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>12.1</b> Compositionality</a></li>
<li class="chapter" data-level="12.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>12.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="core-microbiota" class="section level1">
<h1><span class="header-section-number">6</span> Core microbiota</h1>
<p>For more information:</p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S1198743X14609629?via%3Dihub">The adult intestinal core microbiota is determined by analysis depth and health status</a>.</p>
<p><a href="https://academic.oup.com/femsre/article/41/2/182/2979411">Intestinal microbiome landscaping: insight in community assemblage and implications for microbial modulation strategies</a>.</p>
<p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0023035">Intestinal Microbiota in Healthy Adults: Temporal Analysis Reveals Individual and Common Core and Relation to Intestinal Symptoms</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome) <span class="co"># data analysis and visualisation</span>
<span class="kw">library</span>(phyloseq) <span class="co"># also the basis of data object. Data analysis and visualisation</span>
<span class="kw">library</span>(RColorBrewer) <span class="co"># nice color options</span>
<span class="kw">library</span>(ggpubr) <span class="co"># publication quality figures, based on ggplot2</span>
<span class="kw">library</span>(dplyr) <span class="co"># data handling  </span></code></pre></div>
<div id="core-microbiota-anlaysis" class="section level2">
<h2><span class="header-section-number">6.1</span> Core microbiota anlaysis</h2>
<p>We will use the filtered phyloseq object from previous tutorial. We will use the filtered phyloseq object from the first section for pre-processioning.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># read non rarefied data</span>
ps1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps.ng.tax.rds&quot;</span>)

<span class="co"># use print option to see the data saved as phyloseq object.</span></code></pre></div>
<p>Subset the data to keep only stool samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.stool &lt;-<span class="st"> </span><span class="kw">subset_samples</span>(ps1, bodysite <span class="op">==</span><span class="st"> &quot;Stool&quot;</span>)

<span class="co"># convert to relative abundance  </span>
ps1.stool.rel &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">transform</span>(ps1.stool, <span class="st">&quot;compositional&quot;</span>)
<span class="kw">print</span>(ps1.stool.rel)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 4710 taxa and 169 samples ]
## sample_data() Sample Data:       [ 169 samples by 30 sample variables ]
## tax_table()   Taxonomy Table:    [ 4710 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 4710 tips and 4709 internal nodes ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.stool.rel2 &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps1.stool.rel) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, ps1.stool.rel)

<span class="kw">print</span>(ps1.stool.rel2)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1996 taxa and 169 samples ]
## sample_data() Sample Data:       [ 169 samples by 30 sample variables ]
## tax_table()   Taxonomy Table:    [ 1996 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1996 tips and 1995 internal nodes ]</code></pre>
<p>Check for the core ASVs</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">core.taxa.standard &lt;-<span class="st"> </span><span class="kw">core_members</span>(ps1.stool.rel2, <span class="dt">detection =</span> <span class="fl">0.001</span>, <span class="dt">prevalence =</span> <span class="dv">50</span><span class="op">/</span><span class="dv">100</span>)

<span class="kw">print</span>(core.taxa.standard)</code></pre></div>
<pre><code>##  [1] &quot;94104936&quot;  &quot;941049119&quot; &quot;94104948&quot;  &quot;94104953&quot;  &quot;94104940&quot; 
##  [6] &quot;94104937&quot;  &quot;941049150&quot; &quot;94104955&quot;  &quot;941049451&quot; &quot;94104943&quot; 
## [11] &quot;941049144&quot; &quot;94104962&quot;  &quot;94104956&quot;  &quot;94104959&quot;  &quot;94104974&quot; 
## [16] &quot;941049143&quot; &quot;941049112&quot; &quot;94104958&quot;  &quot;941049125&quot; &quot;941049109&quot;
## [21] &quot;94104969&quot;  &quot;94104960&quot;  &quot;94104968&quot;  &quot;941049102&quot; &quot;941049320&quot;
## [26] &quot;94104946&quot;</code></pre>
<p>There are 16 ASVs that are core based on the cut-offs for prevalence and detection we choose. However, we only see IDs, not very informative. We can get the classification of these as below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Extract the taxonomy table</span>

taxonomy &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">tax_table</span>(ps1.stool.rel2))

<span class="co"># Subset this taxonomy table to include only core OTUs  </span>
core_taxa_id &lt;-<span class="st"> </span><span class="kw">subset</span>(taxonomy, <span class="kw">rownames</span>(taxonomy) <span class="op">%in%</span><span class="st"> </span>core.taxa.standard)

DT<span class="op">::</span><span class="kw">datatable</span>(core_taxa_id)</code></pre></div>
<div id="htmlwidget-ac82dc5448dd2cc486f0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ac82dc5448dd2cc486f0">{"x":{"filter":"none","data":[["94104936","941049119","94104948","94104953","94104940","94104937","941049150","94104955","941049451","94104943","941049144","94104962","94104956","94104959","94104974","941049143","941049112","94104958","941049125","941049109","94104969","94104960","94104968","941049102","941049320","94104946"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes"],["Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia","Clostridia"],["Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales"],["Bacteroidaceae","Bacteroidaceae","Porphyromonadaceae","Porphyromonadaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Porphyromonadaceae","Rikenellaceae","Rikenellaceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Lachnospiraceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae"],["Bacteroides","Bacteroides","Parabacteroides","Parabacteroides","Bacteroides","Bacteroides","Bacteroides","Odoribacter","Alistipes","Alistipes","[Eubacterium]","[Eubacterium]","[Ruminococcus]","Blautia","","Lachnospiraceae","","","Fusicatenibacter","Roseburia","Roseburia","Faecalibacterium","Faecalibacterium","Subdoligranulum","Ruminococcaceae","uncultured"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Domain<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="core-abundance-and-diversity" class="section level2">
<h2><span class="header-section-number">6.2</span> Core abundance and diversity</h2>
<p>Total core abundance in each sample (sum of abundances of the core members):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">core.abundance &lt;-<span class="st"> </span><span class="kw">sample_sums</span>(<span class="kw">core</span>(ps1.stool.rel2, <span class="dt">detection =</span> <span class="fl">0.001</span>, <span class="dt">prevalence =</span> <span class="dv">50</span><span class="op">/</span><span class="dv">100</span>))

DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(core.abundance))</code></pre></div>
<div id="htmlwidget-162f1c21d6220adbc0a5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-162f1c21d6220adbc0a5">{"x":{"filter":"none","data":[["1927.SRS015389.SRX020546.SRR043661","1927.SRS014923.SRX020546.SRR043675","1927.SRS014823.SRX020546.SRR043699","1927.SRS015854.SRX020554.SRR043727","1927.SRS015190.SRX020554.SRR043731","1927.SRS015281.SRX020554.SRR043805","1927.SRS014885.SRX020554.SRR043822","1927.SRS014369.SRX020554.SRR043823","1927.SRS046341.SRX020554.SRR043828","1927.SRS019787.SRX020577.SRR043974","1927.SRS015578.SRX020577.SRR044021","1927.SRS014999.SRX020577.SRR044050","1927.SRS016267.SRX020577.SRR044054","1927.SRS048870.SRX020577.SRR044064","1927.SRS015332.SRX020577.SRR044071","1927.SRS015247.SRX020577.SRR044079","1927.SRS015960.SRX020577.SRR044087","1927.SRS015724.SRX020539.SRR044132","1927.SRS015782.SRX020539.SRR044139","1927.SRS015518.SRX020539.SRR044146","1927.SRS015599.SRX020539.SRR044148","1927.SRS016018.SRX020539.SRR044156","1927.SRS015663.SRX020539.SRR044161","1927.SRS016095.SRX020516.SRR044317","1927.SRS016203.SRX020516.SRR044343","1927.SRS019910.SRX020516.SRR044347","1927.SRS018559.SRX020516.SRR044352","1927.SRS019267.SRX020516.SRR044360","1927.SRS015911.SRX020510.SRR044585","1927.SRS019013.SRX020510.SRR044597","1927.SRS015911.SRX020520.SRR044615","1927.SRS019013.SRX020520.SRR044627","1927.SRS015815.SRX020562.SRR044705","1927.SRS019089.SRX020564.SRR044731","1927.SRS015452.SRX020545.SRR044740","1927.SRS019089.SRX020545.SRR044757","1927.SRS011529.SRX020661.SRR044768","1927.SRS011653.SRX020661.SRR044770","1927.SRS011621.SRX020661.SRR044776","1927.SRS011415.SRX020661.SRR044786","1927.SRS011413.SRX020661.SRR044792","1927.SRS011452.SRX020661.SRR044798","1927.SRS011410.SRX020661.SRR044800","1927.SRS011405.SRX020661.SRR044803","1927.SRS011157.SRX020661.SRR044808","1927.SRS011586.SRX020661.SRR044826","1927.SRS011271.SRX020661.SRR044827","1927.SRS011529.SRX020659.SRR044994","1927.SRS011653.SRX020659.SRR044997","1927.SRS011621.SRX020659.SRR045004","1927.SRS011415.SRX020659.SRR045013","1927.SRS011413.SRX020659.SRR045020","1927.SRS011410.SRX020659.SRR045027","1927.SRS011157.SRX020659.SRR045036","1927.SRS011586.SRX020659.SRR045050","1927.SRS011271.SRX020659.SRR045051","1927.SRS014442.SRX020543.SRR045553","1927.SRS014659.SRX020543.SRR045556","1927.SRS014287.SRX020543.SRR045558","1927.SRS019582.SRX020543.SRR045562","1927.SRS048819.SRX020543.SRR045568","1927.SRS050422.SRX020543.SRR045575","1927.SRS048819.SRX020511.SRR045612","1927.SRS019582.SRX020511.SRR045623","1927.SRS050422.SRX020511.SRR045628","1927.SRS014659.SRX020511.SRR045649","1927.SRS014442.SRX020511.SRR045656","1927.SRS014287.SRX020511.SRR045664","1927.SRS018920.SRX020572.SRR045810","1927.SRS018968.SRX020572.SRR045817","1927.SRS018712.SRX020572.SRR046003","1927.SRS043804.SRX020531.SRR046050","1927.SRS046349.SRX020531.SRR046051","1927.SRS053335.SRX020531.SRR046056","1927.SRS052326.SRX020670.SRR046232","1927.SRS011405.SRX020659.SRR046237","1927.SRS054928.SRX020670.SRR046262","1927.SRS043411.SRX020670.SRR046345","1927.SRS064276.SRX020670.SRR046347","1927.SRS065263.SRX020670.SRR046351","1927.SRS045526.SRX020670.SRR046355","1927.SRS045877.SRX020523.SRR046406","1927.SRS043299.SRX020523.SRR046410","1927.SRS057447.SRX020523.SRR046411","1927.SRS055563.SRX020523.SRR046412","1927.SRS049823.SRX020523.SRR046414","1927.SRS042387.SRX020523.SRR046416","1927.SRS048838.SRX020523.SRR046418","1927.SRS045607.SRX020523.SRR046421","1927.SRS054590.SRX020550.SRR046427","1927.SRS046313.SRX020550.SRR046437","1927.SRS052697.SRX020550.SRR046441","1927.SRS054461.SRX020550.SRR046442","1927.SRS064321.SRX020519.SRR046501","1927.SRS063324.SRX020519.SRR046503","1927.SRS064557.SRX020569.SRR046542","1927.SRS063921.SRX020569.SRR046605","1927.SRS063275.SRX020532.SRR046899","1927.SRS063961.SRX020532.SRR046907","1927.SRS062464.SRX020532.SRR046982","1927.SRS063827.SRX020532.SRR046984","1927.SRS063214.SRX020532.SRR046986","1927.SRS062847.SRX020532.SRR047036","1927.SRS065466.SRX020532.SRR047056","1927.SRS056259.SRX020578.SRR047235","1927.SRS051031.SRX020578.SRR047244","1927.SRS046382.SRX020540.SRR047262","1927.SRS042703.SRX020540.SRR047267","1927.SRS049164.SRX020540.SRR047273","1927.SRS042669.SRX020540.SRR047279","1927.SRS042628.SRX020578.SRR047303","1927.SRS014345.SRX020546.SRR047368","1927.SRS014613.SRX020546.SRR047386","1927.SRS018733.SRX020516.SRR047425","1927.SRS015452.SRX020564.SRR047456","1927.SRS011452.SRX020659.SRR047485","1927.SRS046639.SRX020670.SRR047659","1927.SRS049428.SRX020670.SRR047663","1927.SRS045414.SRX020563.SRR047697","1927.SRS058416.SRX020523.SRR047701","1927.SRS055665.SRX020523.SRR047702","1927.SRS049748.SRX020523.SRR047703","1927.SRS050733.SRX020550.SRR047710","1927.SRS055482.SRX020550.SRR047716","1927.SRS012191.SRX020675.SRR047747","1927.SRS021601.SRX020675.SRR047748","1927.SRS022987.SRX020675.SRR047766","1927.SRS021910.SRX020675.SRR047783","1927.SRS021541.SRX020675.SRR047797","1927.SRS022924.SRX020675.SRR047811","1927.SRS023047.SRX020675.SRR047826","1927.SRS020119.SRX020676.SRR047844","1927.SRS020176.SRX020676.SRR047859","1927.SRS023422.SRX020676.SRR047874","1927.SRS024511.SRX020676.SRR047887","1927.SRS023851.SRX020676.SRR047925","1927.SRS048083.SRX020677.SRR047956","1927.SRS057122.SRX020677.SRR047969","1927.SRS013543.SRX020677.SRR047980","1927.SRS013762.SRX020677.SRR048007","1927.SRS023488.SRX020678.SRR048028","1927.SRS050599.SRX020678.SRR048036","1927.SRS023422.SRX020683.SRR048341","1927.SRS016056.SRX020536.SRR049357","1927.SRS065665.SRX020519.SRR049404","1927.SRS065725.SRX020519.SRR049530","1927.SRS018872.SRX020572.SRR051576","1927.SRS019381.SRX020572.SRR051585","1927.SRS019534.SRX020572.SRR051591","1927.SRS021109.SRX022097.SRR057663","1927.SRS020584.SRX022097.SRR057663","1927.SRS020811.SRX022097.SRR057663","1927.SRS020470.SRX022097.SRR057663","1927.SRS020641.SRX022097.SRR057663","1927.SRS063807.SRX022232.SRR058094","1927.SRS021853.SRX022232.SRR058094","1927.SRS063307.SRX022232.SRR058094","1927.SRS063918.SRX022232.SRR058094","1927.SRS065421.SRX022232.SRR058094","1927.SRS064647.SRX022232.SRR058094","1927.SRS042572.SRX022241.SRR058097","1927.SRS049949.SRX022241.SRR058097","1927.SRS049766.SRX022241.SRR058097","1927.SRS056832.SRX022241.SRR058097","1927.SRS057676.SRX022156.SRR058115","1927.SRS063968.SRX022156.SRR058115","1927.SRS051892.SRX022156.SRR058115","1927.SRS042803.SRX022156.SRR058115","1927.SRS063608.SRX022156.SRR058115"],[0.526949826130154,0.414548108502615,0.643651014170603,0.656386311973951,0.735081267054218,0.540961522269934,0.712558474271321,0.60062488513141,0.486176836861768,0.608977978543196,0.612723515140624,0.378732009040086,0.479402497374256,0.622824974411464,0.56270985581671,0.480575176589304,0.512215282880235,0.605486336509266,0.346825172186055,0.6135800860159,0.560200153964588,0.568763023299867,0.358738501971091,0.288553431798436,0.537527593818985,0.420257778935035,0.549789083471814,0.637278106508876,0.704630428530342,0.647196809758386,0.681742372216227,0.634650455927052,0.323540894007418,0.497500192292901,0.372772506364268,0.518009478672986,0.186523640908144,0.285854737618287,0.517813130181492,0.164796718542647,0.404062465873103,0.808173076923077,0.469072894550602,0.467334035827187,0.242564284867875,0.383274100399822,0.28333099677555,0.191780821917808,0.278895705521472,0.516944284893739,0.176887075483019,0.400043015377998,0.47298631682889,0.251582089552239,0.384649316277018,0.282201086956522,0.699894115867494,0.41240253411306,0.67073732718894,0.396097715736041,0.407373897308301,0.680694455556444,0.436917562724014,0.403284962525913,0.725872179703807,0.407273711758867,0.730653611612127,0.698825428135928,0.243455087004233,0.247287449392713,0.601553306660391,0.483657304138025,0.224004063565779,0.492137574495584,0.788212799460128,0.475863525932186,0.583642063265888,0.692279465727286,0.669820245979186,0.588530016999195,0.562505275597198,0.535652173913044,0.531892021022456,0.496123091834262,0.709915917657292,0.489773820981713,0.154774637127578,0.941359696641387,0.579493613138686,0.810770256752251,0.639719142078809,0.384094368340944,0.36540824541333,0.587568651381745,0.633474819318573,0.63148223764802,0.933384897998646,0.29592549476135,0.476757791724285,0.333919056796282,0.718654086620439,0.22491382612026,0.649083215796897,0.579841240539044,0.494631944062104,0.351998591301285,0.631604973156259,0.562947543713572,0.590926478463775,0.661865569272977,0.434061930783242,0.538324022346369,0.75,0.247405966277562,0.355100693298118,0.830479659987857,0.550978464837816,0.581738614448895,0.590740022362686,0.515222984562607,0.538443199427822,0.208929452392699,0.414466574031187,0.332072943634464,0.321056172216616,0.546559823155568,0.462182528607312,0.192224287484511,0.446036036036036,0.33275645059786,0.424191750278707,0.368755364806867,0.374364504698814,0.355296790684465,0.70170522776399,0.617200129324281,0.696687240206575,0.1950638666378,0.388383045525903,0.729663744312507,0.349028840494408,0.497075409836066,0.374162096282754,0.144327351838422,0.2528076171875,0.533286980705719,0.456405327474942,0.578031212484994,0.660912258473234,0.623556581986143,0.546695402298851,0.287531806615776,0.435498089920658,0.704185386616221,0.24312,0.440685510794569,0.274738633600778,0.828752642706131,0.491824302661109,0.351255151742226,0.250573602221954,0.464883491959304,0.762379851563451,0.223989493652415,0.122714856062198,0.219352351580571,0.243864634461379,0.0978127734033246,0.170414421111318]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>core.abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="core-visualization" class="section level2">
<h2><span class="header-section-number">6.3</span> Core visualization</h2>
<div id="core-heatmaps" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Core heatmaps</h3>
<p>This visualization method has been used for instance in <a href="https://academic.oup.com/femsre/article/41/2/182/2979411">Intestinal microbiome landscaping: insight in community assemblage and implications for microbial modulation strategies</a>.</p>
<p>Note that you can order the taxa on the heatmap with the order.taxa argument.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Core with compositionals:</span>
prevalences &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">05</span>, <span class="dv">1</span>, .<span class="dv">05</span>)
detections &lt;-<span class="st"> </span><span class="dv">10</span><span class="op">^</span><span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-3</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)

<span class="co"># Also define gray color palette</span>
gray &lt;-<span class="st"> </span><span class="kw">gray</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="dv">5</span>))
p.core &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.stool.rel2, 
                    <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, 
                    <span class="dt">colours =</span> gray,
                    <span class="dt">prevalences =</span> prevalences, 
                    <span class="dt">detections =</span> detections, 
                    <span class="dt">min.prevalence =</span> .<span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)
<span class="kw">print</span>(p.core)    </code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Same with the viridis color palette</span>
<span class="co"># color-blind friendly and uniform</span>
<span class="co"># options: viridis, magma, plasma, inferno</span>
<span class="co"># https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html</span>
<span class="co"># Also discrete=TRUE versions available</span>
<span class="kw">library</span>(viridis)</code></pre></div>
<pre><code>## Loading required package: viridisLite</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(p.core <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_viridis</span>())</code></pre></div>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-7-2.png" width="672" /></p>
<p>Color change</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Core with compositionals:</span>
prevalences &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">05</span>, <span class="dv">1</span>, .<span class="dv">05</span>)
detections &lt;-<span class="st"> </span><span class="dv">10</span><span class="op">^</span><span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-3</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)

<span class="co"># Also define gray color palette</span>

p.core &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.stool.rel2, 
                    <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, 
                    <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),
                    <span class="dt">prevalences =</span> prevalences, 
                    <span class="dt">detections =</span> detections, 
                    <span class="dt">min.prevalence =</span> .<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)

<span class="kw">print</span>(p.core) </code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Use the <code>format_to_besthit</code> function from microbiomeutilities to get the best classification of the ASVs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.stool.rel2.f &lt;-<span class="st"> </span>microbiomeutilities<span class="op">::</span><span class="kw">format_to_besthit</span>(ps1.stool.rel2)

p.core &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.stool.rel2.f, 
                    <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, 
                    <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),
                    <span class="dt">prevalences =</span> prevalences, 
                    <span class="dt">detections =</span> detections, 
                    <span class="dt">min.prevalence =</span> .<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)

p.core <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&quot;italic&quot;</span>))</code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(p.core)</code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-9-2.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 16299)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] methods   stats     graphics  grDevices utils     datasets  base     
## 
## other attached packages:
##  [1] bindrcpp_0.2.2       viridis_0.5.1        viridisLite_0.3.0   
##  [4] dplyr_0.7.5          ggpubr_0.1.6         magrittr_1.5        
##  [7] RColorBrewer_1.1-2   microbiome_1.1.10013 ggplot2_2.2.1.9000  
## [10] phyloseq_1.22.3     
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.38.0             tidyr_0.8.1               
##  [3] jsonlite_1.5               splines_3.4.4             
##  [5] foreach_1.4.4              shiny_1.1.0               
##  [7] assertthat_0.2.0           stats4_3.4.4              
##  [9] ggrepel_0.8.0              yaml_2.1.19               
## [11] pillar_1.2.2               backports_1.1.2           
## [13] lattice_0.20-35            glue_1.2.0                
## [15] digest_0.6.15              promises_1.0.1            
## [17] XVector_0.18.0             colorspace_1.3-2          
## [19] htmltools_0.3.6            httpuv_1.4.3              
## [21] Matrix_1.2-12              plyr_1.8.4                
## [23] microbiomeutilities_0.99.0 pkgconfig_2.0.1           
## [25] pheatmap_1.0.10            bookdown_0.7              
## [27] zlibbioc_1.24.0            xtable_1.8-2              
## [29] purrr_0.2.4                scales_0.5.0              
## [31] later_0.7.2                Rtsne_0.13                
## [33] tibble_1.4.2               mgcv_1.8-23               
## [35] IRanges_2.12.0             DT_0.4                    
## [37] withr_2.1.2                BiocGenerics_0.24.0       
## [39] lazyeval_0.2.1             mime_0.5                  
## [41] survival_2.41-3            evaluate_0.10.1           
## [43] nlme_3.1-131.1             MASS_7.3-49               
## [45] vegan_2.5-2                tools_3.4.4               
## [47] data.table_1.11.2          stringr_1.3.1             
## [49] S4Vectors_0.16.0           munsell_0.4.3             
## [51] cluster_2.0.6              Biostrings_2.46.0         
## [53] ade4_1.7-11                compiler_3.4.4            
## [55] rlang_0.2.0                rhdf5_2.22.0              
## [57] grid_3.4.4                 iterators_1.0.9           
## [59] biomformat_1.6.0           htmlwidgets_1.2           
## [61] crosstalk_1.0.0            igraph_1.2.1              
## [63] labeling_0.3               rmarkdown_1.9             
## [65] gtable_0.2.0               codetools_0.2-15          
## [67] multtest_2.34.0            reshape2_1.4.3            
## [69] R6_2.2.2                   gridExtra_2.3             
## [71] knitr_1.20                 bindr_0.1.1               
## [73] rprojroot_1.3-2            permute_0.9-4             
## [75] ape_5.1                    stringi_1.2.2             
## [77] parallel_3.4.4             Rcpp_0.12.17              
## [79] tidyselect_0.2.4           xfun_0.1</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="beta-diversity-metrics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="inference-of-microbial-ecological-networks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
