<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-26">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="inference-of-microbial-ecological-networks.html">
<link rel="next" href="linear-models-the-role-of-covariates.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.5</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.6</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.7</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>2.8</b> Variablity</a></li>
<li class="chapter" data-level="2.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>2.9</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stragey"><i class="fa fa-check"></i><b>2.10</b> NG-tax stragey</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>3</b> Alpha diversities</a><ul>
<li class="chapter" data-level="3.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>3.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="3.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>3.2</b> Diversities</a><ul>
<li class="chapter" data-level="3.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>3.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="3.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>3.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>4</b> Composition plots</a><ul>
<li class="chapter" data-level="4.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>4.1</b> Barplot counts</a></li>
<li class="chapter" data-level="4.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="4.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>4.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>4.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>5</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="5.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>5.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="5.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>5.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>5.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="5.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>5.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="5.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>6</b> Core microbiota</a><ul>
<li class="chapter" data-level="6.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>6.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="6.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>6.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="6.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>6.3</b> Core visualization</a><ul>
<li class="chapter" data-level="6.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>6.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>7</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="7.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>7.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="7.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>7.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="7.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>7.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="7.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>7.2.2</b> igraph network</a></li>
<li class="chapter" data-level="7.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>7.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>7.3</b> Network properties</a><ul>
<li class="chapter" data-level="7.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>7.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="7.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>7.3.2</b> Network plot</a></li>
<li class="chapter" data-level="7.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>7.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>8</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="8.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>8.1</b> Load example data</a></li>
<li class="chapter" data-level="8.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="8.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="8.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>8.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="8.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>8.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>9</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="9.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>9.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="9.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>9.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="9.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>9.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="10.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>10.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="10.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>10.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="10.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>10.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="10.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>10.1.3</b> Rarity</a></li>
<li class="chapter" data-level="10.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>10.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>10.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="10.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>10.3</b> DESeq2: differential abundance testing for sequencing data</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>11</b> Further exercises</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-abundance-testing-for-univariate-data" class="section level1">
<h1><span class="header-section-number">8</span> Differential abundance testing for univariate data</h1>
<p>This section covers basic univariate tests for two-group comparison, covering t-test, Wilcoxon test, and multiple testing. You can try out the suggested exercises in the hands-on session. These are followed by example solutions which we will cover in more detail in the class.</p>
<div id="load-example-data" class="section level2">
<h2><span class="header-section-number">8.1</span> Load example data</h2>
<p>The following example compares the abundance of a selected bug between two conditions. We assume that the data is already properly normalized.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome)
<span class="kw">theme_set</span>(<span class="kw">theme_bw</span>(<span class="dv">20</span>))

<span class="kw">data</span>(dietswap)
d &lt;-<span class="st"> </span>dietswap

<span class="co"># Pick microbial abundances for a given taxonomic group</span>
taxa &lt;-<span class="st"> &quot;Dialister&quot;</span>

<span class="co"># Construct a data.frame with the selected</span>
<span class="co"># taxonomic group and grouping</span>
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Abundance =</span> <span class="kw">abundances</span>(d)[taxa,],
                 <span class="dt">Group =</span> <span class="kw">meta</span>(d)<span class="op">$</span>nationality)

<span class="kw">library</span>(knitr)
<span class="kw">kable</span>(<span class="kw">head</span>(df))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Abundance</th>
<th align="left">Group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Sample-1</td>
<td align="right">310</td>
<td align="left">AAM</td>
</tr>
<tr class="even">
<td>Sample-2</td>
<td align="right">1441</td>
<td align="left">AFR</td>
</tr>
<tr class="odd">
<td>Sample-3</td>
<td align="right">356</td>
<td align="left">AFR</td>
</tr>
<tr class="even">
<td>Sample-4</td>
<td align="right">1528</td>
<td align="left">AFR</td>
</tr>
<tr class="odd">
<td>Sample-5</td>
<td align="right">361</td>
<td align="left">AFR</td>
</tr>
<tr class="even">
<td>Sample-6</td>
<td align="right">2375</td>
<td align="left">AFR</td>
</tr>
</tbody>
</table>
</div>
<div id="visual-comparison-of-two-groups" class="section level2">
<h2><span class="header-section-number">8.2</span> Visual comparison of two groups</h2>
<p><strong>Task: Compare the groups visually</strong> Tips: boxplot, density plot, histogram</p>
<p>Visualization of the absolute abundances is shown on the left. Let us try the log10 transformation. Now, the data contains many zeros and taking log10 will yield infinite values. Hence we choose the commonly used, although somewhat problematic, log10(1+x) transformation (right).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> Abundance)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">       </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Absolute abundances&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Abundance (read count)&quot;</span>)

<span class="co"># Let us add the log10(1+x) version:</span>
df<span class="op">$</span>Log10_Abundance &lt;-<span class="st"> </span><span class="kw">log10</span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>df<span class="op">$</span>Abundance)
p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> Log10_Abundance)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">       </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Log10 abundances&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Abundance (log10(1+x) read count)&quot;</span>)       

<span class="kw">library</span>(patchwork)
p1 <span class="op">+</span><span class="st"> </span>p2</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/univariate_boxplot-1.png" width="600px" /></p>
</div>
<div id="statistical-comparison-of-two-groups" class="section level2">
<h2><span class="header-section-number">8.3</span> Statistical comparison of two groups</h2>
<p><strong>Task: Test whether abundance differences are statistically significant between the two groups</strong> Tips: t-test (t.test); Wilcoxon test (wilcox.test). Find information on how to use by typing help(t.test) or help(wilcox.test); or by looking for examples from the web.</p>
<p>The groups seem to differ. First, let us perform the t-test. This is based on Gaussian assumptions. Each group is expected to follow Gaussian distribution.</p>
<p>Significance p-value with t-test:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">t.test</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value)</code></pre></div>
<pre><code>## [1] 0.06032956</code></pre>
<p>According to this, the abundances is not significantly different between the two groups (at <span class="math inline">\(p&lt;0.05\)</span> level).</p>
</div>
<div id="investigate-assumptions-of-the-t-test" class="section level2">
<h2><span class="header-section-number">8.4</span> Investigate assumptions of the t-test</h2>
<p><strong>Task: Assess whether the abundance data is Gaussian or log-normal within each group</strong> You can use for instance histogram (hist) or density plots (plot(density())).</p>
<p>Now let us investigate the Gaussian assumption of the t-test in more detail. Let us try another visualization; the density plot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">fill =</span> Group, <span class="dt">x =</span> Log10_Abundance)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_density</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>)
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/univariate_densityplot-1.png" width="672" /></p>
<p>Apparently, the data is not even approximately Gaussian distributed. In such cases, a common procedure is to use non-parametric tests. These do not make assumptions of the data distribution but instead compare the ordering of the samples.</p>
<p>So, let us look at the significance p-value with Wilcoxon test (log10 data):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">wilcox.test</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value)</code></pre></div>
<pre><code>## [1] 0.01803111</code></pre>
<p>But since the test is non-parametric, we can as well use the original absolute abundances; thelog transformation does not change sample ordering on which the Wilcoxon test is based.</p>
<p>Let us verify that the absolute abundances yield the same p-value for Wilcoxon test:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">wilcox.test</span>(Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value)</code></pre></div>
<pre><code>## [1] 0.01803111</code></pre>
</div>
<div id="compare-results-between-parametric-and-non-parametric-tests" class="section level2">
<h2><span class="header-section-number">8.5</span> Compare results between parametric and non-parametric tests</h2>
<p>Let us compare how much the results would differ in the whole data between t-test (parametric) and Wilcoxon test (non-parametric).To remove non-varying taxa that would demand extra scripting, let us for demonstration purposes now focus on core taxa that are observed in more than 20% of the samples with more than 3 reads.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Core taxa to be tested</span>
test.taxa &lt;-<span class="st"> </span><span class="kw">core_members</span>(d, <span class="dt">prevalence =</span> <span class="dv">20</span><span class="op">/</span><span class="dv">100</span>, <span class="dt">detection =</span> <span class="dv">3</span>)

<span class="co"># Calculate p-values with the two different methods for each taxonomic unit</span>
pvalue.ttest &lt;-<span class="st"> </span><span class="kw">c</span>()
pvalue.wilcoxon &lt;-<span class="st"> </span><span class="kw">c</span>()
<span class="cf">for</span> (taxa <span class="cf">in</span> test.taxa) {
  <span class="co"># Create a new data frame for each taxonomic group</span>
  df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Abundance =</span> <span class="kw">abundances</span>(d)[taxa,],
                   <span class="dt">Log10_Abundance =</span> <span class="kw">log10</span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw">abundances</span>(d)[taxa,]),  
                   <span class="dt">Group =</span> <span class="kw">meta</span>(d)<span class="op">$</span>nationality)

  pvalue.ttest[[taxa]] &lt;-<span class="st"> </span><span class="kw">t.test</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value
  pvalue.wilcoxon[[taxa]] &lt;-<span class="st"> </span><span class="kw">wilcox.test</span>(Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value  
}
<span class="co"># Arrange the results in a data.frame</span>
pvalues &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">taxon =</span> test.taxa,
                  <span class="dt">pvalue.ttest =</span> pvalue.ttest,
                      <span class="dt">pvalue.wilcoxon =</span> pvalue.wilcoxon)

<span class="co"># Note that multiple testing occurs.</span>
<span class="co"># We must correct the p-values.</span>
<span class="co"># let us apply the standard Benjamini-Hochberg False Discovery Rate (FDR)</span>
<span class="co"># correction</span>
pvalues<span class="op">$</span>pvalue.ttest.adjusted &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(pvalues<span class="op">$</span>pvalue.ttest)
pvalues<span class="op">$</span>pvalue.wilcoxon.adjusted &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(pvalues<span class="op">$</span>pvalue.wilcoxon)</code></pre></div>
<p>Compare the distribution of raw and adjusteed p-values.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pvalues, <span class="kw">aes</span>(<span class="dt">x =</span> pvalue.wilcoxon)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_histogram</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Raw p-values&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">80</span>))

p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pvalues, <span class="kw">aes</span>(<span class="dt">x =</span> pvalue.wilcoxon.adjusted)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_histogram</span>() <span class="op">+</span>
<span class="st">        </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Adjusted p-values&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">80</span>))  

<span class="kw">print</span>(p1 <span class="op">+</span><span class="st"> </span>p2)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/univariate5-1.png" width="600px" /></p>
<p>Now compare these adjusted p-values between t-test and Wilcoxon test. Let us also highlight the p = 0.05 intervals.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> pvalues,
         <span class="kw">aes</span>(<span class="dt">x =</span> pvalue.ttest.adjusted,
         <span class="dt">y =</span> pvalue.wilcoxon.adjusted)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> taxon)) <span class="op">+</span><span class="st"> </span>
<span class="st">       </span><span class="kw">geom_abline</span>(<span class="kw">aes</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope =</span> <span class="dv">1</span>)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_hline</span>(<span class="kw">aes</span>(<span class="dt">yintercept =</span> <span class="fl">0.05</span>), <span class="dt">shape =</span> <span class="dv">2</span>) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_vline</span>(<span class="kw">aes</span>(<span class="dt">xintercept =</span> <span class="fl">0.05</span>), <span class="dt">shape =</span> <span class="dv">2</span>)
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/univariate6-1.png" width="600px" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="inference-of-microbial-ecological-networks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="linear-models-the-role-of-covariates.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
