<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Inference of Microbial Ecological Networks | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</title>
  <meta name="description" content="9 Inference of Microbial Ecological Networks | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Inference of Microbial Ecological Networks | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Inference of Microbial Ecological Networks | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes" />


<meta name="date" content="2020-04-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="differential-abundance-testing-for-univariate-data.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.12/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><i class="fa fa-check"></i><b>2</b> The version 3 of this tutorial from Apr-11-2020 has been tested using</a><ul>
<li class="chapter" data-level="2.1" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#useful-functionsresources"><i class="fa fa-check"></i><b>2.1</b> Useful functions/resources</a></li>
<li class="chapter" data-level="2.2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#focus"><i class="fa fa-check"></i><b>2.2</b> Focus</a></li>
<li class="chapter" data-level="2.3" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#target-audience"><i class="fa fa-check"></i><b>2.3</b> Target audience</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i><b>3</b> Citation</a><ul>
<li class="chapter" data-level="3.1" data-path="citation.html"><a href="citation.html#license"><i class="fa fa-check"></i><b>3.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="4.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>4.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="4.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>4.2</b> General overview</a></li>
<li class="chapter" data-level="4.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>4.3</b> Structure</a></li>
<li class="chapter" data-level="4.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>4.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="4.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>4.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="4.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>4.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="4.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>4.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="4.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>4.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>4.9</b> Variablity</a></li>
<li class="chapter" data-level="4.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>4.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>4.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>5</b> Alpha diversities</a><ul>
<li class="chapter" data-level="5.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>5.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="5.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>5.2</b> Diversities</a><ul>
<li class="chapter" data-level="5.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>5.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="5.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>5.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>6</b> Composition plots</a><ul>
<li class="chapter" data-level="6.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>6.1</b> Barplot counts</a></li>
<li class="chapter" data-level="6.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>6.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="6.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>6.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>6.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>7</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="7.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>7.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>7.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>8</b> Core microbiota</a><ul>
<li class="chapter" data-level="8.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>8.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="8.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>8.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="8.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>8.3</b> Core visualization</a><ul>
<li class="chapter" data-level="8.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>8.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>9</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="9.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>9.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="9.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>9.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="9.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>9.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="9.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>9.2.2</b> igraph network</a></li>
<li class="chapter" data-level="9.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>9.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>9.3</b> Network properties</a><ul>
<li class="chapter" data-level="9.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>9.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="9.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>9.3.2</b> Network plot</a></li>
<li class="chapter" data-level="9.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>9.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>10</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="10.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>10.1</b> Load example data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="10.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="10.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>10.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="10.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>10.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>11</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="11.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>11.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="11.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>11.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="11.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>11.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="12.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>12.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="12.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>12.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="12.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>12.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="12.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>12.1.3</b> Rarity</a></li>
<li class="chapter" data-level="12.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>12.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>12.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="12.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>12.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="12.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>12.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="12.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>12.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>13</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="13.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>13.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="13.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>13.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>14</b> Further exercises</a><ul>
<li class="chapter" data-level="14.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>14.1</b> Compositionality</a></li>
<li class="chapter" data-level="14.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>14.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>15</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="inference-of-microbial-ecological-networks" class="section level1">
<h1><span class="header-section-number">9</span> Inference of Microbial Ecological Networks</h1>
<p>More information on <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226">SPIEC-EASI</a>.</p>
<p>The input for SPIEC-EASI is a counts table. The normalization and tranformation is done by the function.<br />
This step is heavy on computational memory and slow. Noise filtered OTU-OTU level covariance would be ideal.</p>
<p><strong>Load packages and data</strong></p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb207-1" title="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb207-2" title="2"><span class="kw">install_github</span>(<span class="st">&quot;zdk123/SpiecEasi&quot;</span>)</a>
<a class="sourceLine" id="cb207-3" title="3"></a>
<a class="sourceLine" id="cb207-4" title="4"><span class="co"># Other packages you need to install are</span></a>
<a class="sourceLine" id="cb207-5" title="5"><span class="co"># install.packages(&#39;igraph&#39;)</span></a>
<a class="sourceLine" id="cb207-6" title="6"></a>
<a class="sourceLine" id="cb207-7" title="7"><span class="kw">install.packages</span>(<span class="st">&quot;intergraph&quot;</span>)</a>
<a class="sourceLine" id="cb207-8" title="8"><span class="kw">install.packages</span>(<span class="st">&quot;GGally&quot;</span>)</a>
<a class="sourceLine" id="cb207-9" title="9">devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;briatte/ggnet&quot;</span>)</a>
<a class="sourceLine" id="cb207-10" title="10"></a>
<a class="sourceLine" id="cb207-11" title="11"><span class="kw">install.packages</span>(<span class="st">&quot;network&quot;</span>)</a>
<a class="sourceLine" id="cb207-12" title="12"><span class="kw">install.packages</span>(<span class="st">&quot;ggnetwork&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb208-1" title="1"><span class="kw">library</span>(microbiome) <span class="co"># data analysis and visualisation</span></a>
<a class="sourceLine" id="cb208-2" title="2"><span class="kw">library</span>(phyloseq) <span class="co"># also the basis of data object. Data analysis and visualisation</span></a>
<a class="sourceLine" id="cb208-3" title="3"><span class="kw">library</span>(RColorBrewer) <span class="co"># nice color options</span></a>
<a class="sourceLine" id="cb208-4" title="4"><span class="kw">library</span>(ggpubr) <span class="co"># publication quality figures, based on ggplot2</span></a>
<a class="sourceLine" id="cb208-5" title="5"><span class="kw">library</span>(dplyr) <span class="co"># data handling</span></a>
<a class="sourceLine" id="cb208-6" title="6"><span class="kw">library</span>(SpiecEasi) <span class="co"># Network analysis for sparse compositional data  </span></a>
<a class="sourceLine" id="cb208-7" title="7"><span class="kw">library</span>(network)</a>
<a class="sourceLine" id="cb208-8" title="8"><span class="kw">library</span>(intergraph)</a>
<a class="sourceLine" id="cb208-9" title="9"><span class="co">#devtools::install_github(&quot;briatte/ggnet&quot;)</span></a>
<a class="sourceLine" id="cb208-10" title="10"><span class="kw">library</span>(ggnet)</a>
<a class="sourceLine" id="cb208-11" title="11"><span class="kw">library</span>(igraph)</a></code></pre></div>
<p><strong>Read data</strong></p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb209-1" title="1">ps1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps.ng.tax.rds&quot;</span>)</a></code></pre></div>
<p><strong>Select only stool samples</strong></p>
<p>We will subset our data to include only stool samples.</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb210-1" title="1">ps1.stool &lt;-<span class="st"> </span><span class="kw">subset_samples</span>(ps1, bodysite <span class="op">==</span><span class="st"> &quot;Stool&quot;</span>)</a></code></pre></div>
<p><strong>For testing reduce the number of ASVs</strong></p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb211-1" title="1">ps1.stool.otu &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps1.stool) <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>, ps1.stool)</a>
<a class="sourceLine" id="cb211-2" title="2"></a>
<a class="sourceLine" id="cb211-3" title="3"><span class="co"># Add taxonomic classification to OTU ID</span></a>
<a class="sourceLine" id="cb211-4" title="4">ps1.stool.otu.f &lt;-<span class="st"> </span>microbiomeutilities<span class="op">::</span><span class="kw">format_to_besthit</span>(ps1.stool.otu)</a></code></pre></div>
<pre><code>## Warning: replacing previous import &#39;ggplot2::alpha&#39; by &#39;microbiome::alpha&#39; when
## loading &#39;microbiomeutilities&#39;</code></pre>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb213-1" title="1"><span class="kw">head</span>(<span class="kw">tax_table</span>(ps1.stool.otu))</a></code></pre></div>
<pre><code>## Taxonomy Table:     [6 taxa by 6 taxonomic ranks]:
##            Domain     Phylum          Class         Order          
## 9410491526 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
## 9410491516 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
## 9410492612 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
## 9410491521 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
## 9410491824 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
## 9410491817 &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot;
##            Family           Genus        
## 9410491526 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;
## 9410491516 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;
## 9410492612 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;
## 9410491521 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;
## 9410491824 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;
## 9410491817 &quot;Bacteroidaceae&quot; &quot;Bacteroides&quot;</code></pre>
<p>Check the difference in two phyloseq objects.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb215-1" title="1"><span class="kw">head</span>(<span class="kw">tax_table</span>(ps1.stool.otu.f))</a></code></pre></div>
<div id="prepare-data-for-spieceasi" class="section level2">
<h2><span class="header-section-number">9.1</span> Prepare data for SpiecEasi</h2>
<p>The calculation of SpiecEasi are time consuming. For this tutorial, we will have the necessary input files for SpiecEasi.</p>
<ul>
<li>OTU table<br />
</li>
<li>Taxonomy table</li>
</ul>
<p>We save it as <em>.rds</em> object.</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb216-1" title="1">otu.c &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">otu_table</span>(ps1.stool.otu.f)<span class="op">@</span>.Data) <span class="co">#extract the otu table from phyloseq object</span></a>
<a class="sourceLine" id="cb216-2" title="2"></a>
<a class="sourceLine" id="cb216-3" title="3">tax.c &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">tax_table</span>(ps1.stool.otu.f)<span class="op">@</span>.Data)<span class="co">#extract the taxonomy information</span></a>
<a class="sourceLine" id="cb216-4" title="4"></a>
<a class="sourceLine" id="cb216-5" title="5"><span class="kw">head</span>(tax.c)</a></code></pre></div>
<pre><code>##                              Domain        Phylum       Class         Order
## OTU-9410491526:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
## OTU-9410491516:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
## OTU-9410492612:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
## OTU-9410491521:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
## OTU-9410491824:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
## OTU-9410491817:Bacteroides Bacteria Bacteroidetes Bacteroidia Bacteroidales
##                                    Family       Genus
## OTU-9410491526:Bacteroides Bacteroidaceae Bacteroides
## OTU-9410491516:Bacteroides Bacteroidaceae Bacteroides
## OTU-9410492612:Bacteroides Bacteroidaceae Bacteroides
## OTU-9410491521:Bacteroides Bacteroidaceae Bacteroides
## OTU-9410491824:Bacteroides Bacteroidaceae Bacteroides
## OTU-9410491817:Bacteroides Bacteroidaceae Bacteroides
##                                              best_hit
## OTU-9410491526:Bacteroides OTU-9410491526:Bacteroides
## OTU-9410491516:Bacteroides OTU-9410491516:Bacteroides
## OTU-9410492612:Bacteroides OTU-9410492612:Bacteroides
## OTU-9410491521:Bacteroides OTU-9410491521:Bacteroides
## OTU-9410491824:Bacteroides OTU-9410491824:Bacteroides
## OTU-9410491817:Bacteroides OTU-9410491817:Bacteroides</code></pre>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb218-1" title="1"><span class="co"># use this only for first attempt to run it on server to save time</span></a>
<a class="sourceLine" id="cb218-2" title="2"><span class="co">#saveRDS(otu.c, &quot;input_data/stool.otu.c.rds&quot;)</span></a>
<a class="sourceLine" id="cb218-3" title="3"><span class="co">#saveRDS(tax.c, &quot;input_data/stool.tax.c.rds&quot;)</span></a></code></pre></div>
</div>
<div id="spiec-easi-network-reconstruction" class="section level2">
<h2><span class="header-section-number">9.2</span> SPIEC-EASI network reconstruction</h2>
<p>More information on <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226">SPIEC-EASI</a>.</p>
<p>The input for SPIEC-EASI is a counts table. The normalization and tranformation is done by the function. This is very handy tool.<br />
This step is heavy on computational memory and very slow. For this workshop we have already have the output and will skip this chuck.</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb219-1" title="1"><span class="co"># In practice, use more repetitions</span></a>
<a class="sourceLine" id="cb219-2" title="2"><span class="kw">set.seed</span>(<span class="dv">1244</span>)</a>
<a class="sourceLine" id="cb219-3" title="3">net.c &lt;-<span class="st"> </span><span class="kw">spiec.easi</span>(otu.c, <span class="dt">method=</span><span class="st">&#39;mb&#39;</span>, <span class="dt">icov.select.params=</span><span class="kw">list</span>(<span class="dt">rep.num=</span><span class="dv">50</span>)) <span class="co"># reps have to increases for real data</span></a>
<a class="sourceLine" id="cb219-4" title="4"></a>
<a class="sourceLine" id="cb219-5" title="5"><span class="co"># saveRDS(net.c, &quot;input_data/net.c.rds&quot;)</span></a>
<a class="sourceLine" id="cb219-6" title="6"></a>
<a class="sourceLine" id="cb219-7" title="7"><span class="co">#please use more numebr of rep.num (99 or 999) the paraemters </span></a>
<a class="sourceLine" id="cb219-8" title="8"></a>
<a class="sourceLine" id="cb219-9" title="9"><span class="co">## Create graph object and get edge values  </span></a></code></pre></div>
<p><strong>We have save the output of net.c to save time</strong></p>
<p>The output of <code>spiec.easi</code> is stored in <em>./input_data/</em> as <em>stool.net.c.rds</em>. Read this file in R and follow the steps below.</p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb220-1" title="1"><span class="co"># the PC has low processing power, you can read the otuput created by us present in the input_data folder.</span></a>
<a class="sourceLine" id="cb220-2" title="2"><span class="kw">source</span>(<span class="st">&quot;scripts/symBeta.R&quot;</span>) <span class="co"># load custom function to get weights.</span></a>
<a class="sourceLine" id="cb220-3" title="3">net.c &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;input_data/stool.net.rds&quot;</span>)</a>
<a class="sourceLine" id="cb220-4" title="4"><span class="kw">class</span>(net.c)</a></code></pre></div>
<pre><code>## [1] &quot;select&quot;</code></pre>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb222-1" title="1">n.c &lt;-<span class="st"> </span><span class="kw">symBeta</span>(<span class="kw">getOptBeta</span>(net.c))</a></code></pre></div>
<p><strong>Add names to IDs</strong><br />
We also add abundance values to vertex (nodes).</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb223-1" title="1"><span class="kw">colnames</span>(n.c) &lt;-<span class="st"> </span><span class="kw">rownames</span>(n.c) &lt;-<span class="st"> </span><span class="kw">colnames</span>(otu.c)</a>
<a class="sourceLine" id="cb223-2" title="2"></a>
<a class="sourceLine" id="cb223-3" title="3">vsize &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">apply</span>(otu.c, <span class="dv">2</span>, mean)) <span class="co"># add log abundance as properties of vertex/nodes.</span></a></code></pre></div>
<div id="prepare-data-for-plotting" class="section level3">
<h3><span class="header-section-number">9.2.1</span> Prepare data for plotting</h3>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb224-1" title="1">stool.ig &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(n.c, <span class="dt">mode=</span><span class="st">&#39;undirected&#39;</span>, <span class="dt">add.rownames =</span> <span class="ot">TRUE</span>, <span class="dt">weighted =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb224-2" title="2">stool.ig <span class="co"># we can see all the attributes and weights</span></a></code></pre></div>
<pre><code>## IGRAPH a8db9f0 UNW- 679 2454 -- 
## + attr: name (v/c), TRUE (v/c), weight (e/n)
## + edges from a8db9f0 (vertex names):
##  [1] OTU-9410491526:Bacteroides--OTU-9410491516:Bacteroides   
##  [2] OTU-9410491526:Bacteroides--OTU-9410491518:Bacteroides   
##  [3] OTU-9410491526:Bacteroides--OTU-941049327:Bacteroides    
##  [4] OTU-9410491526:Bacteroides--OTU-941049949:Bacteroides    
##  [5] OTU-9410491526:Bacteroides--OTU-9410491514:Bacteroides   
##  [6] OTU-9410491526:Bacteroides--OTU-9410491513:Bacteroides   
##  [7] OTU-9410491526:Bacteroides--OTU-9410491574:Parasutterella
##  [8] OTU-9410491516:Bacteroides--OTU-9410491522:Bacteroides   
## + ... omitted several edges</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb226-1" title="1"><span class="co">#plot(stool.ig)</span></a></code></pre></div>
<p>set the layout option</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb227-1" title="1"><span class="co"># check what is it?</span></a>
<a class="sourceLine" id="cb227-2" title="2">?layout_with_fr</a></code></pre></div>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb228-1" title="1">coords.fdr =<span class="st"> </span><span class="kw">layout_with_fr</span>(stool.ig)</a></code></pre></div>
</div>
<div id="igraph-network" class="section level3">
<h3><span class="header-section-number">9.2.2</span> igraph network</h3>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb229-1" title="1"><span class="kw">E</span>(stool.ig)[weight <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]<span class="op">$</span>color&lt;-<span class="st">&quot;steelblue&quot;</span> <span class="co">#now color the edges based on their values positive is steelblue</span></a>
<a class="sourceLine" id="cb229-2" title="2"><span class="kw">E</span>(stool.ig)[weight <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>]<span class="op">$</span>color&lt;-<span class="st">&quot;orange&quot;</span>  <span class="co">#now color the edges based on their values</span></a>
<a class="sourceLine" id="cb229-3" title="3"></a>
<a class="sourceLine" id="cb229-4" title="4"><span class="kw">plot</span>(stool.ig, <span class="dt">layout=</span>coords.fdr, <span class="dt">vertex.size =</span> <span class="dv">2</span>, <span class="dt">vertex.label.cex =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>The visualisation can be enhanced using <a href="https://briatte.github.io/ggnet/">ggnet</a> R package.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb230-1" title="1">stool.net &lt;-<span class="st"> </span><span class="kw">asNetwork</span>(stool.ig)</a>
<a class="sourceLine" id="cb230-2" title="2">network<span class="op">::</span><span class="kw">set.edge.attribute</span>(stool.net, <span class="st">&quot;color&quot;</span>, <span class="kw">ifelse</span>(stool.net <span class="op">%e%</span><span class="st"> &quot;weight&quot;</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">&quot;steelblue&quot;</span>, <span class="st">&quot;orange&quot;</span>))</a></code></pre></div>
<p>Start adding taxonomic information.</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb231-1" title="1"><span class="kw">colnames</span>(<span class="kw">tax_table</span>(ps1.stool.otu.f))</a></code></pre></div>
<pre><code>## [1] &quot;Domain&quot;   &quot;Phylum&quot;   &quot;Class&quot;    &quot;Order&quot;    &quot;Family&quot;   &quot;Genus&quot;    &quot;best_hit&quot;</code></pre>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb233-1" title="1">phyla &lt;-<span class="st"> </span><span class="kw">map_levels</span>(<span class="kw">colnames</span>(otu.c), <span class="dt">from =</span> <span class="st">&quot;best_hit&quot;</span>, <span class="dt">to =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="kw">tax_table</span>(ps1.stool.otu.f))</a>
<a class="sourceLine" id="cb233-2" title="2">stool.net <span class="op">%v%</span><span class="st"> &quot;Phylum&quot;</span> &lt;-<span class="st"> </span>phyla</a>
<a class="sourceLine" id="cb233-3" title="3">stool.net <span class="op">%v%</span><span class="st"> &quot;nodesize&quot;</span> &lt;-<span class="st"> </span>vsize</a></code></pre></div>
</div>
<div id="network-plot" class="section level3">
<h3><span class="header-section-number">9.2.3</span> Network plot</h3>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb234-1" title="1">mycolors &lt;-<span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>,<span class="st">&quot;#fb9a99&quot;</span>,<span class="st">&quot;#e31a1c&quot;</span>,<span class="st">&quot;#fdbf6f&quot;</span>,<span class="st">&quot;#ff7f00&quot;</span>,<span class="st">&quot;#cab2d6&quot;</span>,<span class="st">&quot;#6a3d9a&quot;</span>,<span class="st">&quot;#ffff99&quot;</span>,<span class="st">&quot;#b15928&quot;</span>))</a>
<a class="sourceLine" id="cb234-2" title="2"></a>
<a class="sourceLine" id="cb234-3" title="3">p &lt;-<span class="st"> </span><span class="kw">ggnet2</span>(stool.net, <span class="dt">node.color =</span> <span class="st">&quot;Phylum&quot;</span>, </a>
<a class="sourceLine" id="cb234-4" title="4">            <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">node.size =</span> <span class="st">&quot;nodesize&quot;</span>, </a>
<a class="sourceLine" id="cb234-5" title="5">            <span class="dt">label.size =</span> <span class="dv">2</span>, <span class="dt">edge.color =</span> <span class="st">&quot;color&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color=</span><span class="kw">guide_legend</span>(<span class="dt">title=</span><span class="st">&quot;Phylum&quot;</span>), <span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>mycolors</a>
<a class="sourceLine" id="cb234-6" title="6"></a>
<a class="sourceLine" id="cb234-7" title="7">p </a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>This is difficult to interpret. One way is to remove nodes that are connected to few other nodes. We can use degree as a network statisitic.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb235-1" title="1">stl.mb &lt;-<span class="st"> </span><span class="kw">degree.distribution</span>(stool.ig)</a>
<a class="sourceLine" id="cb235-2" title="2"><span class="kw">plot</span>(<span class="dv">0</span><span class="op">:</span>(<span class="kw">length</span>(stl.mb)<span class="op">-</span><span class="dv">1</span>), stl.mb, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">35</span>), <span class="dt">type=</span><span class="st">&#39;b&#39;</span>, </a>
<a class="sourceLine" id="cb235-3" title="3">      <span class="dt">ylab=</span><span class="st">&quot;Frequency&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Degree&quot;</span>, <span class="dt">main=</span><span class="st">&quot;Degree Distributions&quot;</span>)</a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb236-1" title="1"><span class="co"># we will look at only taxa connect more than 10 others</span></a>
<a class="sourceLine" id="cb236-2" title="2">p &lt;-<span class="st"> </span><span class="kw">ggnet2</span>(stool.net, <span class="dt">node.color =</span> <span class="st">&quot;Phylum&quot;</span>, </a>
<a class="sourceLine" id="cb236-3" title="3">            <span class="dt">label =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb236-4" title="4">            <span class="dt">label.size =</span> <span class="dv">3</span>, <span class="dt">edge.color =</span> <span class="st">&quot;color&quot;</span>,</a>
<a class="sourceLine" id="cb236-5" title="5">            <span class="dt">size =</span> <span class="st">&quot;degree&quot;</span>, <span class="dt">size.min =</span> <span class="dv">10</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color=</span><span class="kw">guide_legend</span>(<span class="dt">title=</span><span class="st">&quot;Phylum&quot;</span>), <span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>mycolors</a></code></pre></div>
<pre><code>## size.min removed 521 nodes out of 679</code></pre>
<pre><code>## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb239-1" title="1">p </a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-18-2.png" width="672" /></p>
</div>
</div>
<div id="network-properties" class="section level2">
<h2><span class="header-section-number">9.3</span> Network properties</h2>
<p>Check for the number of positive and negative edges.</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb240-1" title="1">betaMat=<span class="kw">as.matrix</span>(<span class="kw">symBeta</span>(<span class="kw">getOptBeta</span>(net.c)))</a>
<a class="sourceLine" id="cb240-2" title="2"></a>
<a class="sourceLine" id="cb240-3" title="3"><span class="co"># We divide by two since an edge is represented by two entries in the matrix.</span></a>
<a class="sourceLine" id="cb240-4" title="4">positive=<span class="kw">length</span>(betaMat[betaMat<span class="op">&gt;</span><span class="dv">0</span>])<span class="op">/</span><span class="dv">2</span> </a>
<a class="sourceLine" id="cb240-5" title="5"></a>
<a class="sourceLine" id="cb240-6" title="6">negative=<span class="kw">length</span>(betaMat[betaMat<span class="op">&lt;</span><span class="dv">0</span>])<span class="op">/</span><span class="dv">2</span> </a>
<a class="sourceLine" id="cb240-7" title="7"></a>
<a class="sourceLine" id="cb240-8" title="8">total=<span class="kw">length</span>(betaMat[betaMat<span class="op">!=</span><span class="dv">0</span>])<span class="op">/</span><span class="dv">2</span> </a></code></pre></div>
<div id="modularity-in-networks" class="section level3">
<h3><span class="header-section-number">9.3.1</span> Modularity in networks</h3>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb241-1" title="1">net.c</a></code></pre></div>
<pre><code>## Model: Meinshausen &amp; Buhlmann Graph Estimation (mb)
## selection criterion: stars 
## Graph dimension: 679 
## sparsity level 0.01066118</code></pre>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb243-1" title="1">mod.net &lt;-<span class="st"> </span>net.c<span class="op">$</span>refit</a>
<a class="sourceLine" id="cb243-2" title="2"></a>
<a class="sourceLine" id="cb243-3" title="3"><span class="kw">colnames</span>(mod.net) &lt;-<span class="st"> </span><span class="kw">rownames</span>(mod.net) &lt;-<span class="st"> </span><span class="kw">colnames</span>(otu.c)<span class="co">#you can remove this </span></a>
<a class="sourceLine" id="cb243-4" title="4"></a>
<a class="sourceLine" id="cb243-5" title="5">vsize &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">apply</span>(otu.c, <span class="dv">2</span>, mean))<span class="co"># value we may or may not use as vertex.attribute</span></a>
<a class="sourceLine" id="cb243-6" title="6"></a>
<a class="sourceLine" id="cb243-7" title="7">stool.ig.mod &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(mod.net, <span class="dt">mode=</span><span class="st">&#39;undirected&#39;</span>, <span class="dt">add.rownames =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb243-8" title="8"><span class="kw">plot</span>(stool.ig.mod) <span class="co"># we can see all the attributes and weights</span></a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb244-1" title="1">stool.net.mod &lt;-<span class="st"> </span><span class="kw">asNetwork</span>(stool.ig.mod)</a></code></pre></div>
<p>Set vertex attributes. We can color by phyla and set the size of nodes based on log2 abundance.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb245-1" title="1">phyla &lt;-<span class="st"> </span><span class="kw">map_levels</span>(<span class="kw">colnames</span>(otu.c), <span class="dt">from =</span> <span class="st">&quot;best_hit&quot;</span>, <span class="dt">to =</span> <span class="st">&quot;Phylum&quot;</span>, <span class="kw">tax_table</span>(ps1.stool.otu.f))</a>
<a class="sourceLine" id="cb245-2" title="2">stool.net.mod <span class="op">%v%</span><span class="st"> &quot;Phylum&quot;</span> &lt;-<span class="st"> </span>phyla</a>
<a class="sourceLine" id="cb245-3" title="3">stool.net.mod <span class="op">%v%</span><span class="st"> &quot;nodesize&quot;</span> &lt;-<span class="st"> </span>vsize</a></code></pre></div>
</div>
<div id="network-plot-1" class="section level3">
<h3><span class="header-section-number">9.3.2</span> Network plot</h3>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb246-1" title="1">mycolors &lt;-<span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#a6cee3&quot;</span>, <span class="st">&quot;#1f78b4&quot;</span>, <span class="st">&quot;#b2df8a&quot;</span>, <span class="st">&quot;#33a02c&quot;</span>,<span class="st">&quot;#fb9a99&quot;</span>,<span class="st">&quot;#e31a1c&quot;</span>,<span class="st">&quot;#fdbf6f&quot;</span>,<span class="st">&quot;#ff7f00&quot;</span>,<span class="st">&quot;#cab2d6&quot;</span>,<span class="st">&quot;#6a3d9a&quot;</span>,<span class="st">&quot;#ffff99&quot;</span>,<span class="st">&quot;#b15928&quot;</span>))</a>
<a class="sourceLine" id="cb246-2" title="2"></a>
<a class="sourceLine" id="cb246-3" title="3"><span class="co"># check the colorpicker in the addins option in RStudio to interactively select color options.  </span></a>
<a class="sourceLine" id="cb246-4" title="4"></a>
<a class="sourceLine" id="cb246-5" title="5">p &lt;-<span class="st"> </span><span class="kw">ggnet2</span>(stool.net.mod, <span class="dt">node.color =</span> <span class="st">&quot;Phylum&quot;</span>, </a>
<a class="sourceLine" id="cb246-6" title="6">            <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">node.size =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb246-7" title="7">            <span class="dt">label.size =</span> <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color=</span><span class="kw">guide_legend</span>(<span class="dt">title=</span><span class="st">&quot;Phylum&quot;</span>), <span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>mycolors</a>
<a class="sourceLine" id="cb246-8" title="8"></a>
<a class="sourceLine" id="cb246-9" title="9">p </a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Identify modularity in networks.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb247-1" title="1">modules =<span class="kw">cluster_fast_greedy</span>(stool.ig.mod)</a>
<a class="sourceLine" id="cb247-2" title="2"></a>
<a class="sourceLine" id="cb247-3" title="3"><span class="kw">print</span>(modules)</a></code></pre></div>
<pre><code>## IGRAPH clustering fast greedy, groups: 12, mod: 0.47
## + groups:
##   $`1`
##    [1] &quot;OTU-9410492646:Bacteroides&quot;          
##    [2] &quot;OTU-9410492641:Bacteroides&quot;          
##    [3] &quot;OTU-9410492645:Bacteroides&quot;          
##    [4] &quot;OTU-9410491981:Parabacteroides&quot;      
##    [5] &quot;OTU-9410491974:Bacteroides&quot;          
##    [6] &quot;OTU-9410491978:Bacteroides&quot;          
##    [7] &quot;OTU-9410491977:Parabacteroides&quot;      
##    [8] &quot;OTU-9410491976:Bacteroides&quot;          
##    [9] &quot;OTU-9410492922:Bacteroides&quot;          
##   + ... omitted several groups/vertices</code></pre>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb249-1" title="1"><span class="kw">modularity</span>(modules)</a></code></pre></div>
<pre><code>## [1] 0.4725467</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb251-1" title="1"><span class="kw">V</span>(stool.ig.mod)<span class="op">$</span>color=modules<span class="op">$</span>membership</a>
<a class="sourceLine" id="cb251-2" title="2"></a>
<a class="sourceLine" id="cb251-3" title="3"><span class="kw">plot</span>(stool.ig.mod, <span class="dt">col =</span> modules, <span class="dt">vertex.size =</span> <span class="dv">4</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>)</a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb252-1" title="1">stool.net.mod <span class="op">%v%</span><span class="st"> &quot;membership&quot;</span> &lt;-<span class="st"> </span>modules<span class="op">$</span>membership</a>
<a class="sourceLine" id="cb252-2" title="2"></a>
<a class="sourceLine" id="cb252-3" title="3">p &lt;-<span class="st"> </span><span class="kw">ggnet2</span>(stool.net.mod, <span class="dt">node.color =</span> <span class="st">&quot;membership&quot;</span>, </a>
<a class="sourceLine" id="cb252-4" title="4">            <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">node.size =</span> <span class="st">&quot;nodesize&quot;</span>, </a>
<a class="sourceLine" id="cb252-5" title="5">            <span class="dt">label.size =</span> <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">guides</span>(<span class="dt">color=</span><span class="kw">guide_legend</span>(<span class="dt">title=</span><span class="st">&quot;membership&quot;</span>), <span class="dt">size =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span>mycolors</a></code></pre></div>
<pre><code>## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb254-1" title="1">p </a></code></pre></div>
<p><img src="07-MAW-PVI_files/figure-html/unnamed-chunk-23-2.png" width="672" /></p>
<p>Check which OTUs are part of different modules.</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb255-1" title="1">modulesOneIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb255-2" title="2">modulesOneOtus=modules<span class="op">$</span>names[modulesOneIndices]</a>
<a class="sourceLine" id="cb255-3" title="3">modulesTwoIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb255-4" title="4">modulesTwoOtus=modules<span class="op">$</span>names[modulesTwoIndices]</a>
<a class="sourceLine" id="cb255-5" title="5"></a>
<a class="sourceLine" id="cb255-6" title="6">modulesThreeIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb255-7" title="7">modulesThreeOtus=modules<span class="op">$</span>names[modulesThreeIndices]</a>
<a class="sourceLine" id="cb255-8" title="8">modulesFourIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb255-9" title="9">modulesFourOtus=modules<span class="op">$</span>names[modulesFourIndices]</a>
<a class="sourceLine" id="cb255-10" title="10"></a>
<a class="sourceLine" id="cb255-11" title="11">modulesFiveIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">5</span>)</a>
<a class="sourceLine" id="cb255-12" title="12">modulesFiveOtus=modules<span class="op">$</span>names[modulesFiveIndices]</a>
<a class="sourceLine" id="cb255-13" title="13">modulesSixIndices=<span class="kw">which</span>(modules<span class="op">$</span>membership<span class="op">==</span><span class="dv">6</span>)</a>
<a class="sourceLine" id="cb255-14" title="14">modulesSixOtus=modules<span class="op">$</span>names[modulesSixIndices]</a>
<a class="sourceLine" id="cb255-15" title="15"></a>
<a class="sourceLine" id="cb255-16" title="16"><span class="kw">print</span>(modulesOneOtus)</a></code></pre></div>
<pre><code>##  [1] &quot;OTU-9410492646:Bacteroides&quot;          
##  [2] &quot;OTU-9410492641:Bacteroides&quot;          
##  [3] &quot;OTU-9410492645:Bacteroides&quot;          
##  [4] &quot;OTU-9410491981:Parabacteroides&quot;      
##  [5] &quot;OTU-9410491974:Bacteroides&quot;          
##  [6] &quot;OTU-9410491978:Bacteroides&quot;          
##  [7] &quot;OTU-9410491977:Parabacteroides&quot;      
##  [8] &quot;OTU-9410491976:Bacteroides&quot;          
##  [9] &quot;OTU-9410492922:Bacteroides&quot;          
## [10] &quot;OTU-9410492638:__&quot;                   
## [11] &quot;OTU-9410492639:__&quot;                   
## [12] &quot;OTU-941049940:Butyricimonas&quot;         
## [13] &quot;OTU-9410491975:Alistipes&quot;            
## [14] &quot;OTU-9410491971:Rikenellaceae&quot;        
## [15] &quot;OTU-9410491972:Rikenellaceae&quot;        
## [16] &quot;OTU-9410492644:Alloprevotella&quot;       
## [17] &quot;OTU-9410491758:Alloprevotella&quot;       
## [18] &quot;OTU-9410491980:Prevotella&quot;           
## [19] &quot;OTU-9410492643:Ruminococcaceae&quot;      
## [20] &quot;OTU-9410492640:Ruminococcaceae&quot;      
## [21] &quot;OTU-9410492921:Phascolarctobacterium&quot;
## [22] &quot;OTU-9410492647:Phascolarctobacterium&quot;
## [23] &quot;OTU-941049538:Sutterella&quot;            
## [24] &quot;OTU-941049552:uncultured&quot;            
## [25] &quot;OTU-9410492642:uncultured&quot;</code></pre>
</div>
<div id="good-reads-for-ecological-networks" class="section level3">
<h3><span class="header-section-number">9.3.3</span> Good reads for ecological networks</h3>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3260507/">Using network analysis to explore co-occurrence patterns in soil microbial communities</a></p>
<p><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002606">Microbial Co-occurrence Relationships in the Human Microbiome</a></p>
<p><a href="http://www.nature.com/ismej/journal/v10/n7/full/ismej2015235a.html">Correlation detection strategies in microbial data sets vary widely in sensitivity and precision</a></p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb257-1" title="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.6.3 (2020-02-29)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 18363)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
## [3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
## [5] LC_TIME=English_Netherlands.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] scales_1.1.0         sna_2.5              statnet.common_4.3.0
##  [4] igraph_1.2.4.2       ggnet_0.1.0          intergraph_2.0-2    
##  [7] network_1.16.0       SpiecEasi_1.0.7      dplyr_0.8.5         
## [10] ggpubr_0.2.5         magrittr_1.5         RColorBrewer_1.1-2  
## [13] microbiome_1.8.0     ggplot2_3.3.0        phyloseq_1.30.0     
## 
## loaded via a namespace (and not attached):
##  [1] viridis_0.5.1               Biobase_2.46.0             
##  [3] tidyr_1.0.2                 viridisLite_0.3.0          
##  [5] VGAM_1.1-2                  jsonlite_1.6.1             
##  [7] splines_3.6.3               foreach_1.4.8              
##  [9] assertthat_0.2.1            stats4_3.6.3               
## [11] yaml_2.2.1                  ggrepel_0.8.2              
## [13] pillar_1.4.3                lattice_0.20-40            
## [15] glue_1.3.2                  digest_0.6.25              
## [17] ggsignif_0.6.0              XVector_0.26.0             
## [19] colorspace_1.4-1            htmltools_0.4.0            
## [21] Matrix_1.2-18               plyr_1.8.6                 
## [23] microbiomeutilities_0.99.02 pkgconfig_2.0.3            
## [25] pheatmap_1.0.12             pulsar_0.3.6               
## [27] bookdown_0.18               zlibbioc_1.32.0            
## [29] purrr_0.3.3                 Rtsne_0.15                 
## [31] huge_1.3.4                  tibble_2.1.3               
## [33] mgcv_1.8-31                 farver_2.0.3               
## [35] IRanges_2.20.2              ellipsis_0.3.0             
## [37] withr_2.1.2                 BiocGenerics_0.32.0        
## [39] survival_3.1-11             crayon_1.3.4               
## [41] evaluate_0.14               nlme_3.1-145               
## [43] MASS_7.3-51.5               vegan_2.5-6                
## [45] tools_3.6.3                 data.table_1.12.8          
## [47] formatR_1.7                 lifecycle_0.2.0            
## [49] stringr_1.4.0               Rhdf5lib_1.8.0             
## [51] S4Vectors_0.24.3            munsell_0.5.0              
## [53] cluster_2.1.0               Biostrings_2.54.0          
## [55] ade4_1.7-15                 compiler_3.6.3             
## [57] rlang_0.4.5                 rhdf5_2.30.1               
## [59] grid_3.6.3                  iterators_1.0.12           
## [61] biomformat_1.14.0           rmarkdown_2.1              
## [63] gtable_0.3.0                codetools_0.2-16           
## [65] multtest_2.42.0             reshape2_1.4.3             
## [67] R6_2.4.1                    gridExtra_2.3              
## [69] knitr_1.28                  permute_0.9-5              
## [71] ape_5.3                     stringi_1.4.6              
## [73] parallel_3.6.3              Rcpp_1.0.4                 
## [75] vctrs_0.2.4                 coda_0.19-3                
## [77] tidyselect_1.0.0            xfun_0.12</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance-testing-for-univariate-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
