<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13 Multivariate comparisons of microbial community composition | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</title>
  <meta name="description" content="13 Multivariate comparisons of microbial community composition | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="13 Multivariate comparisons of microbial community composition | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13 Multivariate comparisons of microbial community composition | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes" />


<meta name="date" content="2020-04-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="advanced-models-for-differential-abundance.html"/>
<link rel="next" href="further-exercises.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.12/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><i class="fa fa-check"></i><b>2</b> The version 3 of this tutorial from Apr-11-2020 has been tested using</a><ul>
<li class="chapter" data-level="2.1" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#useful-functionsresources"><i class="fa fa-check"></i><b>2.1</b> Useful functions/resources</a></li>
<li class="chapter" data-level="2.2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#focus"><i class="fa fa-check"></i><b>2.2</b> Focus</a></li>
<li class="chapter" data-level="2.3" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#target-audience"><i class="fa fa-check"></i><b>2.3</b> Target audience</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i><b>3</b> Citation</a><ul>
<li class="chapter" data-level="3.1" data-path="citation.html"><a href="citation.html#license"><i class="fa fa-check"></i><b>3.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="4.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>4.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="4.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>4.2</b> General overview</a></li>
<li class="chapter" data-level="4.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>4.3</b> Structure</a></li>
<li class="chapter" data-level="4.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>4.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="4.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>4.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="4.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>4.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="4.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>4.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="4.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>4.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>4.9</b> Variablity</a></li>
<li class="chapter" data-level="4.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>4.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>4.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>5</b> Alpha diversities</a><ul>
<li class="chapter" data-level="5.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>5.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="5.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>5.2</b> Diversities</a><ul>
<li class="chapter" data-level="5.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>5.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="5.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>5.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>6</b> Composition plots</a><ul>
<li class="chapter" data-level="6.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>6.1</b> Barplot counts</a></li>
<li class="chapter" data-level="6.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>6.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="6.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>6.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>6.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>7</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="7.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>7.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>7.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>8</b> Core microbiota</a><ul>
<li class="chapter" data-level="8.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>8.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="8.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>8.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="8.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>8.3</b> Core visualization</a><ul>
<li class="chapter" data-level="8.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>8.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>9</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="9.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>9.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="9.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>9.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="9.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>9.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="9.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>9.2.2</b> igraph network</a></li>
<li class="chapter" data-level="9.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>9.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>9.3</b> Network properties</a><ul>
<li class="chapter" data-level="9.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>9.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="9.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>9.3.2</b> Network plot</a></li>
<li class="chapter" data-level="9.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>9.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>10</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="10.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>10.1</b> Load example data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="10.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="10.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>10.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="10.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>10.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>11</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="11.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>11.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="11.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>11.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="11.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>11.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="12.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>12.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="12.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>12.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="12.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>12.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="12.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>12.1.3</b> Rarity</a></li>
<li class="chapter" data-level="12.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>12.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>12.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="12.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>12.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="12.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>12.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="12.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>12.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>13</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="13.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>13.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="13.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>13.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>14</b> Further exercises</a><ul>
<li class="chapter" data-level="14.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>14.1</b> Compositionality</a></li>
<li class="chapter" data-level="14.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>14.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>15</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multivariate-comparisons-of-microbial-community-composition" class="section level1">
<h1><span class="header-section-number">13</span> Multivariate comparisons of microbial community composition</h1>
<p>The above examples focus on comparison per individual taxonomic group. Often, the groups are correlated and we are interested in comparing the overall community composition.</p>
<div id="permanova-1" class="section level2">
<h2><span class="header-section-number">13.1</span> PERMANOVA</h2>
<p>Permutational multivariate analysis of variance <a href="https://onlinelibrary.wiley.com/doi/10.1002/9781118445112.stat07841">further reading</a>. See also <a href="https://www.statmethods.net/stats/anova.html">statmethods</a>.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb295-1" title="1"><span class="kw">library</span>(vegan)</a></code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.5-6</code></pre>
<pre><code>## 
## Attaching package: &#39;vegan&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:microbiome&#39;:
## 
##     diversity</code></pre>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb301-1" title="1">pseq &lt;-<span class="st"> </span>dietswap</a>
<a class="sourceLine" id="cb301-2" title="2"></a>
<a class="sourceLine" id="cb301-3" title="3"><span class="co"># Pick relative abundances (compositional) and sample metadata </span></a>
<a class="sourceLine" id="cb301-4" title="4">pseq.rel &lt;-<span class="st"> </span>microbiome<span class="op">::</span><span class="kw">transform</span>(pseq, <span class="st">&quot;compositional&quot;</span>)</a>
<a class="sourceLine" id="cb301-5" title="5">otu &lt;-<span class="st"> </span><span class="kw">abundances</span>(pseq.rel)</a>
<a class="sourceLine" id="cb301-6" title="6">meta &lt;-<span class="st"> </span><span class="kw">meta</span>(pseq.rel)</a>
<a class="sourceLine" id="cb301-7" title="7"></a>
<a class="sourceLine" id="cb301-8" title="8"><span class="co"># samples x species as input</span></a>
<a class="sourceLine" id="cb301-9" title="9"><span class="kw">library</span>(vegan)</a>
<a class="sourceLine" id="cb301-10" title="10">permanova &lt;-<span class="st"> </span><span class="kw">adonis</span>(<span class="kw">t</span>(otu) <span class="op">~</span><span class="st"> </span>group,</a>
<a class="sourceLine" id="cb301-11" title="11">               <span class="dt">data =</span> meta, <span class="dt">permutations=</span><span class="dv">99</span>, <span class="dt">method =</span> <span class="st">&quot;bray&quot;</span>)</a>
<a class="sourceLine" id="cb301-12" title="12"></a>
<a class="sourceLine" id="cb301-13" title="13"><span class="co"># P-value</span></a>
<a class="sourceLine" id="cb301-14" title="14"><span class="kw">print</span>(<span class="kw">as.data.frame</span>(permanova<span class="op">$</span>aov.tab)[<span class="st">&quot;group&quot;</span>, <span class="st">&quot;Pr(&gt;F)&quot;</span>])</a></code></pre></div>
<pre><code>## [1] 0.05</code></pre>
</div>
<div id="checking-the-homogeneity-condition-1" class="section level2">
<h2><span class="header-section-number">13.2</span> Checking the homogeneity condition</h2>
<p>Type <code>?betadisper</code> in R console for more information.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb303-1" title="1"><span class="co"># Note the assumption of similar multivariate spread among the groups</span></a>
<a class="sourceLine" id="cb303-2" title="2"><span class="co"># ie. analogous to variance homogeneity</span></a>
<a class="sourceLine" id="cb303-3" title="3"><span class="co"># Here the groups have signif. different spreads and</span></a>
<a class="sourceLine" id="cb303-4" title="4"><span class="co"># permanova result may be potentially explained by that.</span></a>
<a class="sourceLine" id="cb303-5" title="5">dist &lt;-<span class="st"> </span><span class="kw">vegdist</span>(<span class="kw">t</span>(otu))</a>
<a class="sourceLine" id="cb303-6" title="6"><span class="kw">anova</span>(<span class="kw">betadisper</span>(dist, meta<span class="op">$</span>group))</a></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: Distances
##            Df  Sum Sq  Mean Sq F value Pr(&gt;F)
## Groups      2 0.01252 0.006262  0.6649 0.5154
## Residuals 219 2.06254 0.009418</code></pre>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb305-1" title="1"><span class="kw">permutest</span>(<span class="kw">betadisper</span>(dist, meta<span class="op">$</span>group), <span class="dt">pairwise =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##            Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
## Groups      2 0.01252 0.006262 0.6649    999  0.505
## Residuals 219 2.06254 0.009418                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         DI      ED    HE
## DI         0.40100 0.693
## ED 0.44203         0.296
## HE 0.69469 0.30560</code></pre>
<p>We can also check which taxa contribute most to the community differences. Are these same or different compared to DESeq2?</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb307-1" title="1">coef &lt;-<span class="st"> </span><span class="kw">coefficients</span>(permanova)[<span class="st">&quot;group1&quot;</span>,]</a>
<a class="sourceLine" id="cb307-2" title="2">top.coef &lt;-<span class="st"> </span>coef[<span class="kw">rev</span>(<span class="kw">order</span>(<span class="kw">abs</span>(coef)))[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]]</a>
<a class="sourceLine" id="cb307-3" title="3"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">14</span>, <span class="dv">2</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb307-4" title="4"><span class="kw">barplot</span>(<span class="kw">sort</span>(top.coef), <span class="dt">horiz =</span> T, <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">main =</span> <span class="st">&quot;Top taxa&quot;</span>)</a></code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/permanovatop-1.png" width="600px" /></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="advanced-models-for-differential-abundance.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="further-exercises.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
