<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-25">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="otu-picking-stratergy.html">
<link rel="next" href="alpha-diversities.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.5</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.6</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.7</b> Read data from OTU-picking stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="variablity.html"><a href="variablity.html"><i class="fa fa-check"></i><b>3</b> Variablity</a></li>
<li class="chapter" data-level="4" data-path="otu-picking-stratergy.html"><a href="otu-picking-stratergy.html"><i class="fa fa-check"></i><b>4</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="5" data-path="ng-tax-stragey.html"><a href="ng-tax-stragey.html"><i class="fa fa-check"></i><b>5</b> NG-tax stragey</a></li>
<li class="chapter" data-level="6" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>6</b> Alpha diversities</a><ul>
<li class="chapter" data-level="6.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>6.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="6.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>6.2</b> Diversities</a><ul>
<li class="chapter" data-level="6.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>6.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="6.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>6.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>7</b> Composition plots</a><ul>
<li class="chapter" data-level="7.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>7.1</b> Barplot counts</a></li>
<li class="chapter" data-level="7.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>7.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="7.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>7.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>7.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>8</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>8.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>8.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>9</b> Core microbiota</a><ul>
<li class="chapter" data-level="9.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>9.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="9.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>9.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="9.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>9.3</b> Core visualization</a><ul>
<li class="chapter" data-level="9.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>9.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>10</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="10.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>10.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="10.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>10.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="10.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>10.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="10.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>10.2.2</b> igraph network</a></li>
<li class="chapter" data-level="10.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>10.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>10.3</b> Network properties</a><ul>
<li class="chapter" data-level="10.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>10.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="10.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>10.3.2</b> Network plot</a></li>
<li class="chapter" data-level="10.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>10.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance-testing-univariate-data.html"><a href="differential-abundance-testing-univariate-data.html"><i class="fa fa-check"></i><b>11</b> Differential abundance testing: univariate data</a></li>
<li class="chapter" data-level="12" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>12</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="12.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#generalized-linear-models-a-very-brief-overview"><i class="fa fa-check"></i><b>12.1</b> Generalized linear models: a very brief overview</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="advanced-models-of-differential-abundance.html"><a href="advanced-models-of-differential-abundance.html"><i class="fa fa-check"></i><b>13</b> Advanced models of differential abundance</a><ul>
<li class="chapter" data-level="13.1" data-path="advanced-models-of-differential-abundance.html"><a href="advanced-models-of-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>13.1</b> DESeq2: differential abundance testing for sequencing data</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ng-tax-stragey" class="section level1">
<h1><span class="header-section-number">5</span> NG-tax stragey</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1<span class="fl">.1</span>.ngtax &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(ps.ng.tax, Family !=<span class="st"> &quot;Mitochondria&quot;</span>)

<span class="co"># also check how many lost </span>

<span class="kw">ntaxa</span>(ps.ng.tax)-<span class="kw">ntaxa</span>(ps1<span class="fl">.1</span>.ngtax)</code></pre></div>
<pre><code>## [1] 5</code></pre>
<p>There is large different in the mitochondrial sequences detected by OTU-picking approach and NG-Tax approach. Usually, sequence of mitochondrial and chloroplast origin are present is low abundance.</p>
<p>Check how many total reads are there in the NG-tax data set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#total number of reads in the dataset</span>
reads_per_asv &lt;-<span class="st"> </span><span class="kw">taxa_sums</span>(ps1<span class="fl">.1</span>.ngtax)
<span class="kw">print</span>(<span class="kw">sum</span>(reads_per_asv))</code></pre></div>
<pre><code>## [1] 4662048</code></pre>
<p>There are 4662048 reads in the total data set.<br />
How many ASVs are less than 10 reads and how many reads do they contain?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">length</span>(reads_per_asv[reads_per_asv &lt;<span class="st"> </span><span class="dv">10</span>]))</code></pre></div>
<pre><code>## [1] 654</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">sum</span>(reads_per_asv[reads_per_asv &lt;<span class="st"> </span><span class="dv">10</span>]))</code></pre></div>
<pre><code>## [1] 4405</code></pre>
<p>To put this into context; out of the 4705 OTUs, a 654 OTUs contain less than 10 reads, which is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>((<span class="dv">654</span>/<span class="dv">4705</span>)*<span class="dv">100</span>)</code></pre></div>
<pre><code>## [1] 13.90011</code></pre>
<p>OTU picking approach.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#total number of reads in the dataset</span>
reads_per_otu &lt;-<span class="st"> </span><span class="kw">taxa_sums</span>(ps0<span class="fl">.1</span>.otu)
<span class="kw">print</span>(<span class="kw">sum</span>(reads_per_otu))</code></pre></div>
<pre><code>## [1] 2398446</code></pre>
<p>There are 2398446 reads in the total data set.<br />
How many ASVs are less than 10 reads and how many reads do they contain?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">length</span>(reads_per_otu[reads_per_otu &lt;<span class="st"> </span><span class="dv">10</span>]))</code></pre></div>
<pre><code>## [1] 1678</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">sum</span>(reads_per_otu[reads_per_otu &lt;<span class="st"> </span><span class="dv">10</span>]))</code></pre></div>
<pre><code>## [1] 5198</code></pre>
<p>To put this into context; out of the 4405 OTUs, a 654 OTUs contain less than 10 reads, which is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>((<span class="dv">1678</span>/<span class="dv">3690</span>)*<span class="dv">100</span>)</code></pre></div>
<pre><code>## [1] 45.47425</code></pre>
<p>This is a major drawback of the OTU picking strategy. This percent can be lowered with NG_tax, DADA2, Deblur like approaches.</p>
<p>Let us see how many singletons are there?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">taxa_sums</span>(ps0<span class="fl">.1</span>.otu) &lt;=<span class="st"> </span><span class="dv">1</span>))</code></pre></div>
<pre><code>## [1] 613</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">taxa_sums</span>(ps1<span class="fl">.1</span>.ngtax) &lt;=<span class="st"> </span><span class="dv">1</span>))</code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Let us see how many doubletons are there?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">taxa_sums</span>(ps0<span class="fl">.1</span>.otu) ==<span class="st"> </span><span class="dv">2</span>))</code></pre></div>
<pre><code>## [1] 279</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">taxa_sums</span>(ps1<span class="fl">.1</span>.ngtax) ==<span class="st"> </span><span class="dv">2</span>))</code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Let us see how many Singletons and doubletons are there?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">taxa_sums</span>(ps0<span class="fl">.1</span>.otu) &lt;=<span class="st"> </span><span class="dv">2</span>))</code></pre></div>
<pre><code>## [1] 892</code></pre>
<p>Singletons and doubletons</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">round</span>((<span class="dv">892</span>/<span class="dv">3690</span>)*<span class="dv">100</span>, <span class="dt">digits =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## [1] 24.17</code></pre>
<p>24.17% of the OTUs are doubletons or singletons. This is suggests that there can be potentially spurious OTUs.</p>
<p>It is important to understand that depending on origin of sample, the parameters in NG-tax have to be changed. If you expect your samples to have large diversity of which rare taxa are a major fraction, using less stringent parameters is important. Every data has its own properties.</p>
<p>It is commonly observed that a large fraction of taxa are rare. A nice reading for this topic is the review by Michael D. J. Lynch &amp; Josh D. Neufeld <a href="https://www.nature.com/articles/nrmicro3400">Ecology and exploration of the rare biosphere</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 7 x64 (build 7601) Service Pack 1
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=Dutch_Netherlands.1252  LC_CTYPE=Dutch_Netherlands.1252   
## [3] LC_MONETARY=Dutch_Netherlands.1252 LC_NUMERIC=C                      
## [5] LC_TIME=Dutch_Netherlands.1252    
## 
## attached base packages:
## [1] methods   stats     graphics  grDevices utils     datasets  base     
## 
## other attached packages:
##  [1] ape_5.1                    dplyr_0.7.4               
##  [3] data.table_1.11.2          DT_0.4                    
##  [5] ggpubr_0.1.6               magrittr_1.5              
##  [7] RColorBrewer_1.1-2         microbiomeutilities_0.99.0
##  [9] microbiome_1.1.10013       ggplot2_2.2.1.9000        
## [11] phyloseq_1.22.3           
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.38.0      viridis_0.5.1       tidyr_0.8.0        
##  [4] jsonlite_1.5        viridisLite_0.3.0   splines_3.4.4      
##  [7] foreach_1.4.4       shiny_1.0.5         assertthat_0.2.0   
## [10] stats4_3.4.4        yaml_2.1.19         ggrepel_0.8.0      
## [13] pillar_1.2.2        backports_1.1.2     lattice_0.20-35    
## [16] glue_1.2.0          digest_0.6.15       promises_1.0.1     
## [19] XVector_0.18.0      colorspace_1.3-2    httpuv_1.4.3       
## [22] htmltools_0.3.6     Matrix_1.2-14       plyr_1.8.4         
## [25] pkgconfig_2.0.1     pheatmap_1.0.8      bookdown_0.7       
## [28] zlibbioc_1.24.0     xtable_1.8-2        purrr_0.2.4        
## [31] scales_0.5.0        later_0.7.2         Rtsne_0.13         
## [34] tibble_1.4.2        mgcv_1.8-23         IRanges_2.12.0     
## [37] withr_2.1.2         BiocGenerics_0.24.0 lazyeval_0.2.1     
## [40] mime_0.5            survival_2.42-3     evaluate_0.10.1    
## [43] nlme_3.1-137        MASS_7.3-50         vegan_2.5-1        
## [46] tools_3.4.4         stringr_1.3.1       S4Vectors_0.16.0   
## [49] munsell_0.4.3       cluster_2.0.7-1     bindrcpp_0.2.2     
## [52] Biostrings_2.46.0   ade4_1.7-11         compiler_3.4.4     
## [55] rlang_0.2.0         rhdf5_2.22.0        grid_3.4.4         
## [58] iterators_1.0.9     biomformat_1.7.0    htmlwidgets_1.2    
## [61] crosstalk_1.0.0     igraph_1.2.1        labeling_0.3       
## [64] rmarkdown_1.9       gtable_0.2.0        codetools_0.2-15   
## [67] multtest_2.34.0     reshape2_1.4.3      R6_2.2.2           
## [70] gridExtra_2.3       knitr_1.20          bindr_0.1.1        
## [73] rprojroot_1.3-2     permute_0.9-4       stringi_1.1.7      
## [76] parallel_3.4.4      Rcpp_0.12.16        xfun_0.1</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="otu-picking-stratergy.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="alpha-diversities.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
