<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 The version 3 of this tutorial from Apr-11-2020 has been tested using | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</title>
  <meta name="description" content="2 The version 3 of this tutorial from Apr-11-2020 has been tested using | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="2 The version 3 of this tutorial from Apr-11-2020 has been tested using | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 The version 3 of this tutorial from Apr-11-2020 has been tested using | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes" />


<meta name="date" content="2020-04-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="citation.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.12/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><i class="fa fa-check"></i><b>2</b> The version 3 of this tutorial from Apr-11-2020 has been tested using</a><ul>
<li class="chapter" data-level="2.1" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#useful-functionsresources"><i class="fa fa-check"></i><b>2.1</b> Useful functions/resources</a></li>
<li class="chapter" data-level="2.2" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#focus"><i class="fa fa-check"></i><b>2.2</b> Focus</a></li>
<li class="chapter" data-level="2.3" data-path="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html"><a href="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using.html#target-audience"><i class="fa fa-check"></i><b>2.3</b> Target audience</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i><b>3</b> Citation</a><ul>
<li class="chapter" data-level="3.1" data-path="citation.html"><a href="citation.html#license"><i class="fa fa-check"></i><b>3.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="4.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>4.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="4.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>4.2</b> General overview</a></li>
<li class="chapter" data-level="4.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>4.3</b> Structure</a></li>
<li class="chapter" data-level="4.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>4.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="4.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>4.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="4.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>4.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="4.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>4.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="4.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>4.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>4.9</b> Variablity</a></li>
<li class="chapter" data-level="4.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>4.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="4.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>4.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>5</b> Alpha diversities</a><ul>
<li class="chapter" data-level="5.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>5.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="5.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>5.2</b> Diversities</a><ul>
<li class="chapter" data-level="5.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>5.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="5.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>5.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>6</b> Composition plots</a><ul>
<li class="chapter" data-level="6.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>6.1</b> Barplot counts</a></li>
<li class="chapter" data-level="6.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>6.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="6.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>6.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>6.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>7</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="7.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>7.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>7.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>8</b> Core microbiota</a><ul>
<li class="chapter" data-level="8.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>8.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="8.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>8.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="8.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>8.3</b> Core visualization</a><ul>
<li class="chapter" data-level="8.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>8.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>9</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="9.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>9.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="9.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>9.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="9.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>9.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="9.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>9.2.2</b> igraph network</a></li>
<li class="chapter" data-level="9.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>9.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>9.3</b> Network properties</a><ul>
<li class="chapter" data-level="9.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>9.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="9.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>9.3.2</b> Network plot</a></li>
<li class="chapter" data-level="9.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>9.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>10</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="10.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>10.1</b> Load example data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="10.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>10.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="10.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>10.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="10.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>10.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>11</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="11.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>11.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="11.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>11.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="11.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>11.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>12</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="12.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>12.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="12.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>12.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="12.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>12.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="12.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>12.1.3</b> Rarity</a></li>
<li class="chapter" data-level="12.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>12.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>12.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="12.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>12.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="12.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>12.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="12.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>12.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>13</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="13.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>13.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="13.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>13.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>14</b> Further exercises</a><ul>
<li class="chapter" data-level="14.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>14.1</b> Compositionality</a></li>
<li class="chapter" data-level="14.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>14.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>15</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="the-version-3-of-this-tutorial-from-apr-11-2020-has-been-tested-using" class="section level1">
<h1><span class="header-section-number">2</span> The version 3 of this tutorial from Apr-11-2020 has been tested using</h1>
<ul>
<li>R version 3.6.3 <a href="https://cran.r-project.org/bin/windows/base/old/3.5.1/">Download link</a>.<br />
</li>
<li>Rstudio <a href="https://www.rstudio.com/products/rstudio/download/">Download link</a>.</li>
</ul>
<p>*Make sure you have admin rights on the machine you install these softwares to avoid access issues.</p>
<p>Download the master branch of the the github repository for the <a href="https://github.com/mibwurrepo/Microbial-bioinformatics-introductory-course-Material-2018">course</a>.</p>
<img src="images/Gihub_repo_step1.png" title="fig:" alt="microbiome R package" />
</center>
<p>Unzip the <code>Microbial-bioinformatics-introductory-course-Material-2018-master.zip</code> folder. Unzip the <code>humanmicrobiome.zip</code> subfolder within the <code>input_data</code> folder so that the <em>humanmicrobiome.biom</em> file is in <code>input_data</code> folder.<br />
open the <code>microbiometutorials.Rproj</code></p>
<p>The script to set-up the RStudio environment with required package installations can be found in the folder named <code>scripts/setup_microbiome_analysis.R</code>. In the image below, there is description of how to run this code.</p>
<img src="images/RStudio_IDE.png" title="fig:" alt="microbiome R package" />
</center>
<div id="useful-functionsresources" class="section level2">
<h2><span class="header-section-number">2.1</span> Useful functions/resources</h2>
<ul>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/10/r-cheat-sheet-3.pdf">Base R</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/03/rmarkdown-cheatsheet-2.0.pdf">R Markdown</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/01/rstudio-IDE-cheatsheet.pdf">RStudio IDE</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/11/ggplot2-cheatsheet-2.1.pdf">ggplot2</a><br />
</li>
<li><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">phyloseq</a><br />
</li>
<li><a href="http://microbiome.github.io/microbiome/">microbiome</a><br />
</li>
<li><a href="http://www.cookbook-r.com/Graphs/">R graphics cookbook</a><br />
</li>
<li><a href="https://microsud.github.io/Tools-Microbiome-Analysis/">List of R tools for microbiome analysis</a></li>
</ul>
</div>
<div id="focus" class="section level2">
<h2><span class="header-section-number">2.2</span> Focus</h2>
<p>The primary aim is introduce microbial community data analysis. There will be talks and discussion on theory and methodology for analysis of microbial community data. We will cover topics ranging from design of studies, sequencing technologies, importance of controls and standardized DNA processing. Supervised hands-on training covering analyses from raw reads using NG-Tax, downstream analysis in R for exploration and analysis of microbiome sequencing data will be priority. There will be a strong focus on using R, R Studio for moving towards reproducible and open science.</p>
</div>
<div id="target-audience" class="section level2">
<h2><span class="header-section-number">2.3</span> Target audience</h2>
<p>Anyone who plans to or is currently doing high throughput microbial community analysis.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="citation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
