<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-25">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="set-up-and-pre-processing.html">
<link rel="next" href="otu-picking-stratergy.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.5</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.6</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.7</b> Read data from OTU-picking stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="variablity.html"><a href="variablity.html"><i class="fa fa-check"></i><b>3</b> Variablity</a></li>
<li class="chapter" data-level="4" data-path="otu-picking-stratergy.html"><a href="otu-picking-stratergy.html"><i class="fa fa-check"></i><b>4</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="5" data-path="ng-tax-stragey.html"><a href="ng-tax-stragey.html"><i class="fa fa-check"></i><b>5</b> NG-tax stragey</a></li>
<li class="chapter" data-level="6" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>6</b> Alpha diversities</a><ul>
<li class="chapter" data-level="6.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>6.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="6.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>6.2</b> Diversities</a><ul>
<li class="chapter" data-level="6.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>6.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="6.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>6.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>7</b> Composition plots</a><ul>
<li class="chapter" data-level="7.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>7.1</b> Barplot counts</a></li>
<li class="chapter" data-level="7.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>7.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="7.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>7.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>7.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>8</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>8.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>8.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>9</b> Core microbiota</a><ul>
<li class="chapter" data-level="9.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>9.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="9.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>9.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="9.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>9.3</b> Core visualization</a><ul>
<li class="chapter" data-level="9.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>9.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>10</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="10.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>10.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="10.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>10.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="10.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>10.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="10.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>10.2.2</b> igraph network</a></li>
<li class="chapter" data-level="10.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>10.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>10.3</b> Network properties</a><ul>
<li class="chapter" data-level="10.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>10.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="10.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>10.3.2</b> Network plot</a></li>
<li class="chapter" data-level="10.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>10.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-abundance-testing-univariate-data.html"><a href="differential-abundance-testing-univariate-data.html"><i class="fa fa-check"></i><b>11</b> Differential abundance testing: univariate data</a></li>
<li class="chapter" data-level="12" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>12</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="12.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#generalized-linear-models-a-very-brief-overview"><i class="fa fa-check"></i><b>12.1</b> Generalized linear models: a very brief overview</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="advanced-models-of-differential-abundance.html"><a href="advanced-models-of-differential-abundance.html"><i class="fa fa-check"></i><b>13</b> Advanced models of differential abundance</a><ul>
<li class="chapter" data-level="13.1" data-path="advanced-models-of-differential-abundance.html"><a href="advanced-models-of-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>13.1</b> DESeq2: differential abundance testing for sequencing data</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variablity" class="section level1">
<h1><span class="header-section-number">3</span> Variablity</h1>
<p>We will check the spread of variation for taxa identifed using both approaches. Coefficient of variation, i.e. sd(x)/mean(x) is a widely used approach to measure spread in the data.</p>
<p>Now plot C.V.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">plot_taxa_cv</span>(ps.ng.tax, <span class="dt">plot.type =</span> <span class="st">&quot;scatter&quot;</span>)</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p1 +<span class="st"> </span><span class="kw">scale_x_log10</span>()</code></pre></div>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="st"> </span><span class="kw">plot_taxa_cv</span>(ps.otu, <span class="dt">plot.type =</span> <span class="st">&quot;scatter&quot;</span>)</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2+<span class="st"> </span><span class="kw">scale_x_log10</span>()</code></pre></div>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-13-2.png" width="672" /></p>
<p>From the above two plots, we see that there are several OTUs which a have high C.V. The OTU-picking approach seems to be noisy and NG-Tax approach seems to have less noise, although with little high ASVs identified even after filtering for low abundant reads. Thus, the ASVs identified by NG-Tax are of high confidence as they seem to be reprsented by high number of sequences in the raw data.</p>
<p>Let us check for distribution of numer of sequences retained after OTU picking anf NG-tax approach.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_seqdepth.otu &lt;-<span class="st"> </span><span class="kw">plot_read_distribution</span>(ps.otu, <span class="st">&quot;scientific_name&quot;</span>, <span class="st">&quot;density&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;Done plotting&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_seqdepth.ngtax &lt;-<span class="st"> </span><span class="kw">plot_read_distribution</span>(ps.ng.tax, <span class="st">&quot;scientific_name&quot;</span>, <span class="st">&quot;density&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;Done plotting&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(p_seqdepth.otu)</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(p_seqdepth.ngtax)</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>We see a better distribution of reads in NG-Tax data.</p>
<p>Moving on to distribution of taxa</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We make a data table with information on the OTUs</span>
ps0_df_taxa &lt;-<span class="st"> </span><span class="kw">data.table</span>(<span class="kw">tax_table</span>(ps.ng.tax), 
                        <span class="dt">ASVabundance =</span> <span class="kw">taxa_sums</span>(ps.ng.tax), 
                        <span class="dt">ASV=</span> <span class="kw">taxa_names</span>(ps.ng.tax))

ps0_tax_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(ps0_df_taxa, <span class="kw">aes</span>(ASVabundance)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_histogram</span>() +<span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Histogram of ASVs (unique sequence) counts (NG-Tax)&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme_bw</span>() +<span class="st"> </span><span class="kw">scale_x_log10</span>()

<span class="kw">print</span>(ps0_tax_plot)</code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We make a data table with information on the OTUs</span>
ps1_df_taxa &lt;-<span class="st"> </span><span class="kw">data.table</span>(<span class="kw">tax_table</span>(ps.otu), 
                        <span class="dt">OTUabundance =</span> <span class="kw">taxa_sums</span>(ps.otu), 
                        <span class="dt">OTU =</span> <span class="kw">taxa_names</span>(ps.otu))

ps1_tax_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(ps1_df_taxa, <span class="kw">aes</span>(OTUabundance)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_histogram</span>() +<span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Histogram of OTU (unique sequence) counts (OTU-picking 97%)&quot;</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme_bw</span>() +<span class="st"> </span><span class="kw">scale_x_log10</span>()

<span class="kw">print</span>(ps1_tax_plot)</code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>The ASVs distribution is close to normal distribution compared to OTU distribution for 97% otu-picking approach. This has implications in downstream statistical analysis like differential abundance testing (covered on Day3).</p>
<p>We can also check for prevalance abundance distribtuion of ASVs and OTUs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">plot_taxa_prevalence</span>(ps.ng.tax, <span class="st">&quot;Phylum&quot;</span>)
p</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="st"> </span><span class="kw">plot_taxa_prevalence</span>(ps.otu, <span class="st">&quot;Phylum&quot;</span>)
p2</code></pre></div>
<p><img src="02_Comparison_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>The ASVs identified by NG-tax at low prevalance are represented by high abundance in the sequencing data. There is a higher chance that these are real observations.</p>
<p>Checking for potentially spurious OTUs/ASVs. Usually, we do not expect mitochondria and chloroplast as part of the human microbiome.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="set-up-and-pre-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="otu-picking-stratergy.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
